{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "2c8d2215-828b-417e-8009-8765154e75ea",
   "metadata": {},
   "source": [
    "# Summarize PDTC Metadata\n",
    "## Date: 2022-02-15\n",
    "## Author: Jeffrey Grover\n",
    "**Purpose:** Summarize the improvement in the PDTC Metadata."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "149ed2c4-402e-492f-b2d4-ec65c5f2285c",
   "metadata": {},
   "source": [
    "### Load libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "e1f989c0-5e44-476c-b278-5066dac1a000",
   "metadata": {},
   "outputs": [],
   "source": [
    "library(tidyverse)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6d394550-6f34-46ca-9b2f-ef6f2f76f344",
   "metadata": {},
   "source": [
    "### Load the metadata\n",
    "Since it's been fixed. What's the missing data situation look like?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "1b98d6ce-84e9-4f76-944e-ff7e95c4f45a",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "New names:\n",
      "* `` -> ...1\n",
      "\n",
      "\u001b[1mRows: \u001b[22m\u001b[34m2180\u001b[39m \u001b[1mColumns: \u001b[22m\u001b[34m26\u001b[39m\n",
      "\n",
      "\u001b[36m──\u001b[39m \u001b[1mColumn specification\u001b[22m \u001b[36m──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────\u001b[39m\n",
      "\u001b[1mDelimiter:\u001b[22m \",\"\n",
      "\u001b[31mchr\u001b[39m (24): contributor, file_name, patient_id, model_id, tumor_id, case_id, s...\n",
      "\u001b[32mdbl\u001b[39m  (2): ...1, paired_end\n",
      "\n",
      "\n",
      "\u001b[36mℹ\u001b[39m Use \u001b[30m\u001b[47m\u001b[30m\u001b[47m`spec()`\u001b[47m\u001b[30m\u001b[49m\u001b[39m to retrieve the full column specification for this data.\n",
      "\u001b[36mℹ\u001b[39m Specify the column types or set \u001b[30m\u001b[47m\u001b[30m\u001b[47m`show_col_types = FALSE`\u001b[47m\u001b[30m\u001b[49m\u001b[39m to quiet this message.\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "2180"
      ],
      "text/latex": [
       "2180"
      ],
      "text/markdown": [
       "2180"
      ],
      "text/plain": [
       "[1] 2180"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 26</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>...1</th><th scope=col>contributor</th><th scope=col>file_name</th><th scope=col>patient_id</th><th scope=col>model_id</th><th scope=col>tumor_id</th><th scope=col>case_id</th><th scope=col>sample_id</th><th scope=col>sample_type</th><th scope=col>experimental_strategy</th><th scope=col>⋯</th><th scope=col>access_level</th><th scope=col>data_category</th><th scope=col>data_format</th><th scope=col>data_type</th><th scope=col>platform</th><th scope=col>is_ffpe</th><th scope=col>passage</th><th scope=col>body_location</th><th scope=col>disease_type</th><th scope=col>comment</th></tr>\n",
       "\t<tr><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td>BCM</td><td>LZML_37246_NoIndex_L00X_R1_001.fastq.gz</td><td>24452</td><td>BCM-0002</td><td>2011012-2  </td><td>NA</td><td>BCM-0002</td><td>PDX</td><td>RNA-Seq</td><td>⋯</td><td>Controlled</td><td>Raw Sequencing Data</td><td>fastq.gz</td><td>Raw Reads</td><td>Illumina</td><td>FALSE</td><td>Unknown</td><td>Breast</td><td>Invasive breast carcinoma</td><td>NA</td></tr>\n",
       "\t<tr><td>2</td><td>BCM</td><td>LZML_37246_NoIndex_L00X_R2_001.fastq.gz</td><td>24452</td><td>BCM-0002</td><td>2011012-2  </td><td>NA</td><td>BCM-0002</td><td>PDX</td><td>RNA-Seq</td><td>⋯</td><td>Controlled</td><td>Raw Sequencing Data</td><td>fastq.gz</td><td>Raw Reads</td><td>Illumina</td><td>FALSE</td><td>Unknown</td><td>Breast</td><td>Invasive breast carcinoma</td><td>NA</td></tr>\n",
       "\t<tr><td>3</td><td>BCM</td><td>ML6065_GATCAGCG_L00X_R1_001.fastq.gz   </td><td>2219 </td><td>BCM-0046</td><td>2011012-46 </td><td>NA</td><td>BCM-0046</td><td>PDX</td><td>RNA-Seq</td><td>⋯</td><td>Controlled</td><td>Raw Sequencing Data</td><td>fastq.gz</td><td>Raw Reads</td><td>Illumina</td><td>FALSE</td><td>Unknown</td><td>Breast</td><td>Invasive breast carcinoma</td><td>NA</td></tr>\n",
       "\t<tr><td>4</td><td>BCM</td><td>ML6065_GATCAGCG_L00X_R2_001.fastq.gz   </td><td>2219 </td><td>BCM-0046</td><td>2011012-46 </td><td>NA</td><td>BCM-0046</td><td>PDX</td><td>RNA-Seq</td><td>⋯</td><td>Controlled</td><td>Raw Sequencing Data</td><td>fastq.gz</td><td>Raw Reads</td><td>Illumina</td><td>FALSE</td><td>Unknown</td><td>Breast</td><td>Invasive breast carcinoma</td><td>NA</td></tr>\n",
       "\t<tr><td>5</td><td>BCM</td><td>ML6061_CAGATCTG_L00X_R1_001.fastq.gz   </td><td>24561</td><td>BCM-0104</td><td>2011012-104</td><td>NA</td><td>BCM-0104</td><td>PDX</td><td>RNA-Seq</td><td>⋯</td><td>Controlled</td><td>Raw Sequencing Data</td><td>fastq.gz</td><td>Raw Reads</td><td>Illumina</td><td>FALSE</td><td>Unknown</td><td>Breast</td><td>Invasive breast carcinoma</td><td>NA</td></tr>\n",
       "\t<tr><td>6</td><td>BCM</td><td>ML6061_CAGATCTG_L00X_R2_001.fastq.gz   </td><td>24561</td><td>BCM-0104</td><td>2011012-104</td><td>NA</td><td>BCM-0104</td><td>PDX</td><td>RNA-Seq</td><td>⋯</td><td>Controlled</td><td>Raw Sequencing Data</td><td>fastq.gz</td><td>Raw Reads</td><td>Illumina</td><td>FALSE</td><td>Unknown</td><td>Breast</td><td>Invasive breast carcinoma</td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 26\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " ...1 & contributor & file\\_name & patient\\_id & model\\_id & tumor\\_id & case\\_id & sample\\_id & sample\\_type & experimental\\_strategy & ⋯ & access\\_level & data\\_category & data\\_format & data\\_type & platform & is\\_ffpe & passage & body\\_location & disease\\_type & comment\\\\\n",
       " <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & ⋯ & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t 1 & BCM & LZML\\_37246\\_NoIndex\\_L00X\\_R1\\_001.fastq.gz & 24452 & BCM-0002 & 2011012-2   & NA & BCM-0002 & PDX & RNA-Seq & ⋯ & Controlled & Raw Sequencing Data & fastq.gz & Raw Reads & Illumina & FALSE & Unknown & Breast & Invasive breast carcinoma & NA\\\\\n",
       "\t 2 & BCM & LZML\\_37246\\_NoIndex\\_L00X\\_R2\\_001.fastq.gz & 24452 & BCM-0002 & 2011012-2   & NA & BCM-0002 & PDX & RNA-Seq & ⋯ & Controlled & Raw Sequencing Data & fastq.gz & Raw Reads & Illumina & FALSE & Unknown & Breast & Invasive breast carcinoma & NA\\\\\n",
       "\t 3 & BCM & ML6065\\_GATCAGCG\\_L00X\\_R1\\_001.fastq.gz    & 2219  & BCM-0046 & 2011012-46  & NA & BCM-0046 & PDX & RNA-Seq & ⋯ & Controlled & Raw Sequencing Data & fastq.gz & Raw Reads & Illumina & FALSE & Unknown & Breast & Invasive breast carcinoma & NA\\\\\n",
       "\t 4 & BCM & ML6065\\_GATCAGCG\\_L00X\\_R2\\_001.fastq.gz    & 2219  & BCM-0046 & 2011012-46  & NA & BCM-0046 & PDX & RNA-Seq & ⋯ & Controlled & Raw Sequencing Data & fastq.gz & Raw Reads & Illumina & FALSE & Unknown & Breast & Invasive breast carcinoma & NA\\\\\n",
       "\t 5 & BCM & ML6061\\_CAGATCTG\\_L00X\\_R1\\_001.fastq.gz    & 24561 & BCM-0104 & 2011012-104 & NA & BCM-0104 & PDX & RNA-Seq & ⋯ & Controlled & Raw Sequencing Data & fastq.gz & Raw Reads & Illumina & FALSE & Unknown & Breast & Invasive breast carcinoma & NA\\\\\n",
       "\t 6 & BCM & ML6061\\_CAGATCTG\\_L00X\\_R2\\_001.fastq.gz    & 24561 & BCM-0104 & 2011012-104 & NA & BCM-0104 & PDX & RNA-Seq & ⋯ & Controlled & Raw Sequencing Data & fastq.gz & Raw Reads & Illumina & FALSE & Unknown & Breast & Invasive breast carcinoma & NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 26\n",
       "\n",
       "| ...1 &lt;dbl&gt; | contributor &lt;chr&gt; | file_name &lt;chr&gt; | patient_id &lt;chr&gt; | model_id &lt;chr&gt; | tumor_id &lt;chr&gt; | case_id &lt;chr&gt; | sample_id &lt;chr&gt; | sample_type &lt;chr&gt; | experimental_strategy &lt;chr&gt; | ⋯ ⋯ | access_level &lt;chr&gt; | data_category &lt;chr&gt; | data_format &lt;chr&gt; | data_type &lt;chr&gt; | platform &lt;chr&gt; | is_ffpe &lt;chr&gt; | passage &lt;chr&gt; | body_location &lt;chr&gt; | disease_type &lt;chr&gt; | comment &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | BCM | LZML_37246_NoIndex_L00X_R1_001.fastq.gz | 24452 | BCM-0002 | 2011012-2   | NA | BCM-0002 | PDX | RNA-Seq | ⋯ | Controlled | Raw Sequencing Data | fastq.gz | Raw Reads | Illumina | FALSE | Unknown | Breast | Invasive breast carcinoma | NA |\n",
       "| 2 | BCM | LZML_37246_NoIndex_L00X_R2_001.fastq.gz | 24452 | BCM-0002 | 2011012-2   | NA | BCM-0002 | PDX | RNA-Seq | ⋯ | Controlled | Raw Sequencing Data | fastq.gz | Raw Reads | Illumina | FALSE | Unknown | Breast | Invasive breast carcinoma | NA |\n",
       "| 3 | BCM | ML6065_GATCAGCG_L00X_R1_001.fastq.gz    | 2219  | BCM-0046 | 2011012-46  | NA | BCM-0046 | PDX | RNA-Seq | ⋯ | Controlled | Raw Sequencing Data | fastq.gz | Raw Reads | Illumina | FALSE | Unknown | Breast | Invasive breast carcinoma | NA |\n",
       "| 4 | BCM | ML6065_GATCAGCG_L00X_R2_001.fastq.gz    | 2219  | BCM-0046 | 2011012-46  | NA | BCM-0046 | PDX | RNA-Seq | ⋯ | Controlled | Raw Sequencing Data | fastq.gz | Raw Reads | Illumina | FALSE | Unknown | Breast | Invasive breast carcinoma | NA |\n",
       "| 5 | BCM | ML6061_CAGATCTG_L00X_R1_001.fastq.gz    | 24561 | BCM-0104 | 2011012-104 | NA | BCM-0104 | PDX | RNA-Seq | ⋯ | Controlled | Raw Sequencing Data | fastq.gz | Raw Reads | Illumina | FALSE | Unknown | Breast | Invasive breast carcinoma | NA |\n",
       "| 6 | BCM | ML6061_CAGATCTG_L00X_R2_001.fastq.gz    | 24561 | BCM-0104 | 2011012-104 | NA | BCM-0104 | PDX | RNA-Seq | ⋯ | Controlled | Raw Sequencing Data | fastq.gz | Raw Reads | Illumina | FALSE | Unknown | Breast | Invasive breast carcinoma | NA |\n",
       "\n"
      ],
      "text/plain": [
       "  ...1 contributor file_name                               patient_id model_id\n",
       "1 1    BCM         LZML_37246_NoIndex_L00X_R1_001.fastq.gz 24452      BCM-0002\n",
       "2 2    BCM         LZML_37246_NoIndex_L00X_R2_001.fastq.gz 24452      BCM-0002\n",
       "3 3    BCM         ML6065_GATCAGCG_L00X_R1_001.fastq.gz    2219       BCM-0046\n",
       "4 4    BCM         ML6065_GATCAGCG_L00X_R2_001.fastq.gz    2219       BCM-0046\n",
       "5 5    BCM         ML6061_CAGATCTG_L00X_R1_001.fastq.gz    24561      BCM-0104\n",
       "6 6    BCM         ML6061_CAGATCTG_L00X_R2_001.fastq.gz    24561      BCM-0104\n",
       "  tumor_id    case_id sample_id sample_type experimental_strategy ⋯\n",
       "1 2011012-2   NA      BCM-0002  PDX         RNA-Seq               ⋯\n",
       "2 2011012-2   NA      BCM-0002  PDX         RNA-Seq               ⋯\n",
       "3 2011012-46  NA      BCM-0046  PDX         RNA-Seq               ⋯\n",
       "4 2011012-46  NA      BCM-0046  PDX         RNA-Seq               ⋯\n",
       "5 2011012-104 NA      BCM-0104  PDX         RNA-Seq               ⋯\n",
       "6 2011012-104 NA      BCM-0104  PDX         RNA-Seq               ⋯\n",
       "  access_level data_category       data_format data_type platform is_ffpe\n",
       "1 Controlled   Raw Sequencing Data fastq.gz    Raw Reads Illumina FALSE  \n",
       "2 Controlled   Raw Sequencing Data fastq.gz    Raw Reads Illumina FALSE  \n",
       "3 Controlled   Raw Sequencing Data fastq.gz    Raw Reads Illumina FALSE  \n",
       "4 Controlled   Raw Sequencing Data fastq.gz    Raw Reads Illumina FALSE  \n",
       "5 Controlled   Raw Sequencing Data fastq.gz    Raw Reads Illumina FALSE  \n",
       "6 Controlled   Raw Sequencing Data fastq.gz    Raw Reads Illumina FALSE  \n",
       "  passage body_location disease_type              comment\n",
       "1 Unknown Breast        Invasive breast carcinoma NA     \n",
       "2 Unknown Breast        Invasive breast carcinoma NA     \n",
       "3 Unknown Breast        Invasive breast carcinoma NA     \n",
       "4 Unknown Breast        Invasive breast carcinoma NA     \n",
       "5 Unknown Breast        Invasive breast carcinoma NA     \n",
       "6 Unknown Breast        Invasive breast carcinoma NA     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "pdtc_metadata <- read_csv('../2022-02-09_pdxnet_pdtc_pdmr_portal_ids/2022-02-15_pdtc_metadata_cleaned.csv')\n",
    "\n",
    "nrow(pdtc_metadata)\n",
    "head(pdtc_metadata)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "9feaf277-4f0e-4273-a5e4-134a4b6f2ae2",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Define function to summarize missing fields\n",
    "\n",
    "summarize_missing_fields <- function(df) {\n",
    "    colSums((is.na(df) | (df == 'Unknown') | (df == '?'))) %>% as.data.frame()\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "88d4bf15-777f-4fd4-b8e6-27ab72a62ec8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 26 × 1</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>.</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>...1</th><td>   0</td></tr>\n",
       "\t<tr><th scope=row>contributor</th><td>   0</td></tr>\n",
       "\t<tr><th scope=row>file_name</th><td>   0</td></tr>\n",
       "\t<tr><th scope=row>patient_id</th><td>  26</td></tr>\n",
       "\t<tr><th scope=row>model_id</th><td> 870</td></tr>\n",
       "\t<tr><th scope=row>tumor_id</th><td>1422</td></tr>\n",
       "\t<tr><th scope=row>case_id</th><td> 602</td></tr>\n",
       "\t<tr><th scope=row>sample_id</th><td>   0</td></tr>\n",
       "\t<tr><th scope=row>sample_type</th><td>   0</td></tr>\n",
       "\t<tr><th scope=row>experimental_strategy</th><td>   0</td></tr>\n",
       "\t<tr><th scope=row>gender</th><td>  48</td></tr>\n",
       "\t<tr><th scope=row>paired_end</th><td>   0</td></tr>\n",
       "\t<tr><th scope=row>capture_kit</th><td>  26</td></tr>\n",
       "\t<tr><th scope=row>capture_assembly</th><td>1570</td></tr>\n",
       "\t<tr><th scope=row>availability</th><td>   0</td></tr>\n",
       "\t<tr><th scope=row>public</th><td>   0</td></tr>\n",
       "\t<tr><th scope=row>access_level</th><td>   0</td></tr>\n",
       "\t<tr><th scope=row>data_category</th><td>   0</td></tr>\n",
       "\t<tr><th scope=row>data_format</th><td>   0</td></tr>\n",
       "\t<tr><th scope=row>data_type</th><td>   0</td></tr>\n",
       "\t<tr><th scope=row>platform</th><td>   0</td></tr>\n",
       "\t<tr><th scope=row>is_ffpe</th><td>1714</td></tr>\n",
       "\t<tr><th scope=row>passage</th><td>1174</td></tr>\n",
       "\t<tr><th scope=row>body_location</th><td>  80</td></tr>\n",
       "\t<tr><th scope=row>disease_type</th><td>  80</td></tr>\n",
       "\t<tr><th scope=row>comment</th><td>2144</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 26 × 1\n",
       "\\begin{tabular}{r|l}\n",
       "  & .\\\\\n",
       "  & <dbl>\\\\\n",
       "\\hline\n",
       "\t...1 &    0\\\\\n",
       "\tcontributor &    0\\\\\n",
       "\tfile\\_name &    0\\\\\n",
       "\tpatient\\_id &   26\\\\\n",
       "\tmodel\\_id &  870\\\\\n",
       "\ttumor\\_id & 1422\\\\\n",
       "\tcase\\_id &  602\\\\\n",
       "\tsample\\_id &    0\\\\\n",
       "\tsample\\_type &    0\\\\\n",
       "\texperimental\\_strategy &    0\\\\\n",
       "\tgender &   48\\\\\n",
       "\tpaired\\_end &    0\\\\\n",
       "\tcapture\\_kit &   26\\\\\n",
       "\tcapture\\_assembly & 1570\\\\\n",
       "\tavailability &    0\\\\\n",
       "\tpublic &    0\\\\\n",
       "\taccess\\_level &    0\\\\\n",
       "\tdata\\_category &    0\\\\\n",
       "\tdata\\_format &    0\\\\\n",
       "\tdata\\_type &    0\\\\\n",
       "\tplatform &    0\\\\\n",
       "\tis\\_ffpe & 1714\\\\\n",
       "\tpassage & 1174\\\\\n",
       "\tbody\\_location &   80\\\\\n",
       "\tdisease\\_type &   80\\\\\n",
       "\tcomment & 2144\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 26 × 1\n",
       "\n",
       "| <!--/--> | . &lt;dbl&gt; |\n",
       "|---|---|\n",
       "| ...1 |    0 |\n",
       "| contributor |    0 |\n",
       "| file_name |    0 |\n",
       "| patient_id |   26 |\n",
       "| model_id |  870 |\n",
       "| tumor_id | 1422 |\n",
       "| case_id |  602 |\n",
       "| sample_id |    0 |\n",
       "| sample_type |    0 |\n",
       "| experimental_strategy |    0 |\n",
       "| gender |   48 |\n",
       "| paired_end |    0 |\n",
       "| capture_kit |   26 |\n",
       "| capture_assembly | 1570 |\n",
       "| availability |    0 |\n",
       "| public |    0 |\n",
       "| access_level |    0 |\n",
       "| data_category |    0 |\n",
       "| data_format |    0 |\n",
       "| data_type |    0 |\n",
       "| platform |    0 |\n",
       "| is_ffpe | 1714 |\n",
       "| passage | 1174 |\n",
       "| body_location |   80 |\n",
       "| disease_type |   80 |\n",
       "| comment | 2144 |\n",
       "\n"
      ],
      "text/plain": [
       "                      .   \n",
       "...1                     0\n",
       "contributor              0\n",
       "file_name                0\n",
       "patient_id              26\n",
       "model_id               870\n",
       "tumor_id              1422\n",
       "case_id                602\n",
       "sample_id                0\n",
       "sample_type              0\n",
       "experimental_strategy    0\n",
       "gender                  48\n",
       "paired_end               0\n",
       "capture_kit             26\n",
       "capture_assembly      1570\n",
       "availability             0\n",
       "public                   0\n",
       "access_level             0\n",
       "data_category            0\n",
       "data_format              0\n",
       "data_type                0\n",
       "platform                 0\n",
       "is_ffpe               1714\n",
       "passage               1174\n",
       "body_location           80\n",
       "disease_type            80\n",
       "comment               2144"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# How many NAs are there for the most important columns?\n",
    "\n",
    "\n",
    "summarize_missing_fields(pdtc_metadata)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "80322eb9-055f-4566-b4c7-97c2af6f1673",
   "metadata": {},
   "source": [
    "Note: Some of these *should* have NAs in them. For example, \"passage\" is not meaningful for primary tumor samples."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "eb31433b-c5e0-468a-ae95-aeef851839d0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<dl>\n",
       "\t<dt>$BCM</dt>\n",
       "\t\t<dd><table class=\"dataframe\">\n",
       "<caption>A tibble: 1 × 26</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>...1</th><th scope=col>contributor</th><th scope=col>file_name</th><th scope=col>patient_id</th><th scope=col>model_id</th><th scope=col>tumor_id</th><th scope=col>case_id</th><th scope=col>sample_id</th><th scope=col>sample_type</th><th scope=col>experimental_strategy</th><th scope=col>⋯</th><th scope=col>access_level</th><th scope=col>data_category</th><th scope=col>data_format</th><th scope=col>data_type</th><th scope=col>platform</th><th scope=col>is_ffpe</th><th scope=col>passage</th><th scope=col>body_location</th><th scope=col>disease_type</th><th scope=col>comment</th></tr>\n",
       "\t<tr><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td>BCM</td><td>LZML_37246_NoIndex_L00X_R1_001.fastq.gz</td><td>24452</td><td>BCM-0002</td><td>2011012-2</td><td>NA</td><td>BCM-0002</td><td>PDX</td><td>RNA-Seq</td><td>⋯</td><td>Controlled</td><td>Raw Sequencing Data</td><td>fastq.gz</td><td>Raw Reads</td><td>Illumina</td><td>FALSE</td><td>Unknown</td><td>Breast</td><td>Invasive breast carcinoma</td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n",
       "</dd>\n",
       "\t<dt>$HCI</dt>\n",
       "\t\t<dd><table class=\"dataframe\">\n",
       "<caption>A tibble: 1 × 26</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>...1</th><th scope=col>contributor</th><th scope=col>file_name</th><th scope=col>patient_id</th><th scope=col>model_id</th><th scope=col>tumor_id</th><th scope=col>case_id</th><th scope=col>sample_id</th><th scope=col>sample_type</th><th scope=col>experimental_strategy</th><th scope=col>⋯</th><th scope=col>access_level</th><th scope=col>data_category</th><th scope=col>data_format</th><th scope=col>data_type</th><th scope=col>platform</th><th scope=col>is_ffpe</th><th scope=col>passage</th><th scope=col>body_location</th><th scope=col>disease_type</th><th scope=col>comment</th></tr>\n",
       "\t<tr><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>209</td><td>HCI</td><td>14990X12_HCI001_patientnormal_1.txt.gz</td><td>HCI001</td><td>NA</td><td>NA</td><td>HCI001</td><td>14990X12</td><td>Normal</td><td>WES</td><td>⋯</td><td>Controlled</td><td>Raw Sequencing Data</td><td>fastq.gz</td><td>Raw Reads</td><td>Illumina HiSeq</td><td>Unknown</td><td>NA</td><td>Breast</td><td>Breast Poorly Differentiated</td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n",
       "</dd>\n",
       "\t<dt>$MDACC</dt>\n",
       "\t\t<dd><table class=\"dataframe\">\n",
       "<caption>A tibble: 1 × 26</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>...1</th><th scope=col>contributor</th><th scope=col>file_name</th><th scope=col>patient_id</th><th scope=col>model_id</th><th scope=col>tumor_id</th><th scope=col>case_id</th><th scope=col>sample_id</th><th scope=col>sample_type</th><th scope=col>experimental_strategy</th><th scope=col>⋯</th><th scope=col>access_level</th><th scope=col>data_category</th><th scope=col>data_format</th><th scope=col>data_type</th><th scope=col>platform</th><th scope=col>is_ffpe</th><th scope=col>passage</th><th scope=col>body_location</th><th scope=col>disease_type</th><th scope=col>comment</th></tr>\n",
       "\t<tr><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>455</td><td>MDACC</td><td>BFang-LungPDX-MS11DNA-TC211W.R1.fastq.gz</td><td>TC211</td><td>NA</td><td>NA</td><td>TC211</td><td>TC211W</td><td>Normal</td><td>WES</td><td>⋯</td><td>Controlled</td><td>Raw Sequencing Data</td><td>fastq.gz</td><td>Raw Reads</td><td>Illumina Hiseq</td><td>Unknown</td><td>NA</td><td>Blood</td><td>Blood</td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n",
       "</dd>\n",
       "\t<dt>$`UC Davis`</dt>\n",
       "\t\t<dd><table class=\"dataframe\">\n",
       "<caption>A tibble: 1 × 26</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>...1</th><th scope=col>contributor</th><th scope=col>file_name</th><th scope=col>patient_id</th><th scope=col>model_id</th><th scope=col>tumor_id</th><th scope=col>case_id</th><th scope=col>sample_id</th><th scope=col>sample_type</th><th scope=col>experimental_strategy</th><th scope=col>⋯</th><th scope=col>access_level</th><th scope=col>data_category</th><th scope=col>data_format</th><th scope=col>data_type</th><th scope=col>platform</th><th scope=col>is_ffpe</th><th scope=col>passage</th><th scope=col>body_location</th><th scope=col>disease_type</th><th scope=col>comment</th></tr>\n",
       "\t<tr><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>873</td><td>UC Davis</td><td>BL0269F404_P0_GES10_166_ATCACG_L004_R1_ALL.fastq</td><td>BL0269F</td><td>BL0269F</td><td>BL0269F</td><td>BL0269F</td><td>BL0269F</td><td>Tumor</td><td>RNA-Seq</td><td>⋯</td><td>Controlled</td><td>Raw Sequencing Data</td><td>fastq</td><td>Raw Reads</td><td>Illumina</td><td>Unknown</td><td>P0</td><td>Bladder</td><td>bladder urothelial carcinoma</td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n",
       "</dd>\n",
       "\t<dt>$WISTAR</dt>\n",
       "\t\t<dd><table class=\"dataframe\">\n",
       "<caption>A tibble: 1 × 26</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>...1</th><th scope=col>contributor</th><th scope=col>file_name</th><th scope=col>patient_id</th><th scope=col>model_id</th><th scope=col>tumor_id</th><th scope=col>case_id</th><th scope=col>sample_id</th><th scope=col>sample_type</th><th scope=col>experimental_strategy</th><th scope=col>⋯</th><th scope=col>access_level</th><th scope=col>data_category</th><th scope=col>data_format</th><th scope=col>data_type</th><th scope=col>platform</th><th scope=col>is_ffpe</th><th scope=col>passage</th><th scope=col>body_location</th><th scope=col>disease_type</th><th scope=col>comment</th></tr>\n",
       "\t<tr><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>921</td><td>WISTAR</td><td>wm3907n_R1.fastq.gz</td><td>WM3907</td><td>wm3907n</td><td>NA</td><td>NA</td><td>wm3907n</td><td>Tumor</td><td>WES</td><td>⋯</td><td>Open Access</td><td>Raw Sequencing Data</td><td>fastq.gz</td><td>Raw Reads</td><td>Illumina</td><td>FALSE</td><td>NA</td><td>Skin</td><td>Melanoma</td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n",
       "</dd>\n",
       "\t<dt>$WUSTL</dt>\n",
       "\t\t<dd><table class=\"dataframe\">\n",
       "<caption>A tibble: 1 × 26</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>...1</th><th scope=col>contributor</th><th scope=col>file_name</th><th scope=col>patient_id</th><th scope=col>model_id</th><th scope=col>tumor_id</th><th scope=col>case_id</th><th scope=col>sample_id</th><th scope=col>sample_type</th><th scope=col>experimental_strategy</th><th scope=col>⋯</th><th scope=col>access_level</th><th scope=col>data_category</th><th scope=col>data_format</th><th scope=col>data_type</th><th scope=col>platform</th><th scope=col>is_ffpe</th><th scope=col>passage</th><th scope=col>body_location</th><th scope=col>disease_type</th><th scope=col>comment</th></tr>\n",
       "\t<tr><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1303</td><td>WUSTL</td><td>172845-121-T.r1.fastq.gz</td><td>PDMR-172845</td><td>172845-121-T_SL</td><td>121-T</td><td>PDMR-172845</td><td>172845-121-T_SL</td><td>PDX</td><td>RNA-Seq</td><td>⋯</td><td>Open</td><td>Raw Sequencing Data</td><td>fastq.gz</td><td>Raw Reads</td><td>NovaSeq S4</td><td>Unknown</td><td>P3</td><td>Digestive/Gastrointestinal</td><td>Adenocarcinoma - colon</td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n",
       "</dd>\n",
       "</dl>\n"
      ],
      "text/latex": [
       "\\begin{description}\n",
       "\\item[\\$BCM] A tibble: 1 × 26\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " ...1 & contributor & file\\_name & patient\\_id & model\\_id & tumor\\_id & case\\_id & sample\\_id & sample\\_type & experimental\\_strategy & ⋯ & access\\_level & data\\_category & data\\_format & data\\_type & platform & is\\_ffpe & passage & body\\_location & disease\\_type & comment\\\\\n",
       " <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & ⋯ & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t 1 & BCM & LZML\\_37246\\_NoIndex\\_L00X\\_R1\\_001.fastq.gz & 24452 & BCM-0002 & 2011012-2 & NA & BCM-0002 & PDX & RNA-Seq & ⋯ & Controlled & Raw Sequencing Data & fastq.gz & Raw Reads & Illumina & FALSE & Unknown & Breast & Invasive breast carcinoma & NA\\\\\n",
       "\\end{tabular}\n",
       "\n",
       "\\item[\\$HCI] A tibble: 1 × 26\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " ...1 & contributor & file\\_name & patient\\_id & model\\_id & tumor\\_id & case\\_id & sample\\_id & sample\\_type & experimental\\_strategy & ⋯ & access\\_level & data\\_category & data\\_format & data\\_type & platform & is\\_ffpe & passage & body\\_location & disease\\_type & comment\\\\\n",
       " <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & ⋯ & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t 209 & HCI & 14990X12\\_HCI001\\_patientnormal\\_1.txt.gz & HCI001 & NA & NA & HCI001 & 14990X12 & Normal & WES & ⋯ & Controlled & Raw Sequencing Data & fastq.gz & Raw Reads & Illumina HiSeq & Unknown & NA & Breast & Breast Poorly Differentiated & NA\\\\\n",
       "\\end{tabular}\n",
       "\n",
       "\\item[\\$MDACC] A tibble: 1 × 26\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " ...1 & contributor & file\\_name & patient\\_id & model\\_id & tumor\\_id & case\\_id & sample\\_id & sample\\_type & experimental\\_strategy & ⋯ & access\\_level & data\\_category & data\\_format & data\\_type & platform & is\\_ffpe & passage & body\\_location & disease\\_type & comment\\\\\n",
       " <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & ⋯ & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t 455 & MDACC & BFang-LungPDX-MS11DNA-TC211W.R1.fastq.gz & TC211 & NA & NA & TC211 & TC211W & Normal & WES & ⋯ & Controlled & Raw Sequencing Data & fastq.gz & Raw Reads & Illumina Hiseq & Unknown & NA & Blood & Blood & NA\\\\\n",
       "\\end{tabular}\n",
       "\n",
       "\\item[\\$`UC Davis`] A tibble: 1 × 26\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " ...1 & contributor & file\\_name & patient\\_id & model\\_id & tumor\\_id & case\\_id & sample\\_id & sample\\_type & experimental\\_strategy & ⋯ & access\\_level & data\\_category & data\\_format & data\\_type & platform & is\\_ffpe & passage & body\\_location & disease\\_type & comment\\\\\n",
       " <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & ⋯ & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t 873 & UC Davis & BL0269F404\\_P0\\_GES10\\_166\\_ATCACG\\_L004\\_R1\\_ALL.fastq & BL0269F & BL0269F & BL0269F & BL0269F & BL0269F & Tumor & RNA-Seq & ⋯ & Controlled & Raw Sequencing Data & fastq & Raw Reads & Illumina & Unknown & P0 & Bladder & bladder urothelial carcinoma & NA\\\\\n",
       "\\end{tabular}\n",
       "\n",
       "\\item[\\$WISTAR] A tibble: 1 × 26\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " ...1 & contributor & file\\_name & patient\\_id & model\\_id & tumor\\_id & case\\_id & sample\\_id & sample\\_type & experimental\\_strategy & ⋯ & access\\_level & data\\_category & data\\_format & data\\_type & platform & is\\_ffpe & passage & body\\_location & disease\\_type & comment\\\\\n",
       " <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & ⋯ & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t 921 & WISTAR & wm3907n\\_R1.fastq.gz & WM3907 & wm3907n & NA & NA & wm3907n & Tumor & WES & ⋯ & Open Access & Raw Sequencing Data & fastq.gz & Raw Reads & Illumina & FALSE & NA & Skin & Melanoma & NA\\\\\n",
       "\\end{tabular}\n",
       "\n",
       "\\item[\\$WUSTL] A tibble: 1 × 26\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " ...1 & contributor & file\\_name & patient\\_id & model\\_id & tumor\\_id & case\\_id & sample\\_id & sample\\_type & experimental\\_strategy & ⋯ & access\\_level & data\\_category & data\\_format & data\\_type & platform & is\\_ffpe & passage & body\\_location & disease\\_type & comment\\\\\n",
       " <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & ⋯ & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t 1303 & WUSTL & 172845-121-T.r1.fastq.gz & PDMR-172845 & 172845-121-T\\_SL & 121-T & PDMR-172845 & 172845-121-T\\_SL & PDX & RNA-Seq & ⋯ & Open & Raw Sequencing Data & fastq.gz & Raw Reads & NovaSeq S4 & Unknown & P3 & Digestive/Gastrointestinal & Adenocarcinoma - colon & NA\\\\\n",
       "\\end{tabular}\n",
       "\n",
       "\\end{description}\n"
      ],
      "text/markdown": [
       "$BCM\n",
       ":   \n",
       "A tibble: 1 × 26\n",
       "\n",
       "| ...1 &lt;dbl&gt; | contributor &lt;chr&gt; | file_name &lt;chr&gt; | patient_id &lt;chr&gt; | model_id &lt;chr&gt; | tumor_id &lt;chr&gt; | case_id &lt;chr&gt; | sample_id &lt;chr&gt; | sample_type &lt;chr&gt; | experimental_strategy &lt;chr&gt; | ⋯ ⋯ | access_level &lt;chr&gt; | data_category &lt;chr&gt; | data_format &lt;chr&gt; | data_type &lt;chr&gt; | platform &lt;chr&gt; | is_ffpe &lt;chr&gt; | passage &lt;chr&gt; | body_location &lt;chr&gt; | disease_type &lt;chr&gt; | comment &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | BCM | LZML_37246_NoIndex_L00X_R1_001.fastq.gz | 24452 | BCM-0002 | 2011012-2 | NA | BCM-0002 | PDX | RNA-Seq | ⋯ | Controlled | Raw Sequencing Data | fastq.gz | Raw Reads | Illumina | FALSE | Unknown | Breast | Invasive breast carcinoma | NA |\n",
       "\n",
       "\n",
       "$HCI\n",
       ":   \n",
       "A tibble: 1 × 26\n",
       "\n",
       "| ...1 &lt;dbl&gt; | contributor &lt;chr&gt; | file_name &lt;chr&gt; | patient_id &lt;chr&gt; | model_id &lt;chr&gt; | tumor_id &lt;chr&gt; | case_id &lt;chr&gt; | sample_id &lt;chr&gt; | sample_type &lt;chr&gt; | experimental_strategy &lt;chr&gt; | ⋯ ⋯ | access_level &lt;chr&gt; | data_category &lt;chr&gt; | data_format &lt;chr&gt; | data_type &lt;chr&gt; | platform &lt;chr&gt; | is_ffpe &lt;chr&gt; | passage &lt;chr&gt; | body_location &lt;chr&gt; | disease_type &lt;chr&gt; | comment &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 209 | HCI | 14990X12_HCI001_patientnormal_1.txt.gz | HCI001 | NA | NA | HCI001 | 14990X12 | Normal | WES | ⋯ | Controlled | Raw Sequencing Data | fastq.gz | Raw Reads | Illumina HiSeq | Unknown | NA | Breast | Breast Poorly Differentiated | NA |\n",
       "\n",
       "\n",
       "$MDACC\n",
       ":   \n",
       "A tibble: 1 × 26\n",
       "\n",
       "| ...1 &lt;dbl&gt; | contributor &lt;chr&gt; | file_name &lt;chr&gt; | patient_id &lt;chr&gt; | model_id &lt;chr&gt; | tumor_id &lt;chr&gt; | case_id &lt;chr&gt; | sample_id &lt;chr&gt; | sample_type &lt;chr&gt; | experimental_strategy &lt;chr&gt; | ⋯ ⋯ | access_level &lt;chr&gt; | data_category &lt;chr&gt; | data_format &lt;chr&gt; | data_type &lt;chr&gt; | platform &lt;chr&gt; | is_ffpe &lt;chr&gt; | passage &lt;chr&gt; | body_location &lt;chr&gt; | disease_type &lt;chr&gt; | comment &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 455 | MDACC | BFang-LungPDX-MS11DNA-TC211W.R1.fastq.gz | TC211 | NA | NA | TC211 | TC211W | Normal | WES | ⋯ | Controlled | Raw Sequencing Data | fastq.gz | Raw Reads | Illumina Hiseq | Unknown | NA | Blood | Blood | NA |\n",
       "\n",
       "\n",
       "$`UC Davis`\n",
       ":   \n",
       "A tibble: 1 × 26\n",
       "\n",
       "| ...1 &lt;dbl&gt; | contributor &lt;chr&gt; | file_name &lt;chr&gt; | patient_id &lt;chr&gt; | model_id &lt;chr&gt; | tumor_id &lt;chr&gt; | case_id &lt;chr&gt; | sample_id &lt;chr&gt; | sample_type &lt;chr&gt; | experimental_strategy &lt;chr&gt; | ⋯ ⋯ | access_level &lt;chr&gt; | data_category &lt;chr&gt; | data_format &lt;chr&gt; | data_type &lt;chr&gt; | platform &lt;chr&gt; | is_ffpe &lt;chr&gt; | passage &lt;chr&gt; | body_location &lt;chr&gt; | disease_type &lt;chr&gt; | comment &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 873 | UC Davis | BL0269F404_P0_GES10_166_ATCACG_L004_R1_ALL.fastq | BL0269F | BL0269F | BL0269F | BL0269F | BL0269F | Tumor | RNA-Seq | ⋯ | Controlled | Raw Sequencing Data | fastq | Raw Reads | Illumina | Unknown | P0 | Bladder | bladder urothelial carcinoma | NA |\n",
       "\n",
       "\n",
       "$WISTAR\n",
       ":   \n",
       "A tibble: 1 × 26\n",
       "\n",
       "| ...1 &lt;dbl&gt; | contributor &lt;chr&gt; | file_name &lt;chr&gt; | patient_id &lt;chr&gt; | model_id &lt;chr&gt; | tumor_id &lt;chr&gt; | case_id &lt;chr&gt; | sample_id &lt;chr&gt; | sample_type &lt;chr&gt; | experimental_strategy &lt;chr&gt; | ⋯ ⋯ | access_level &lt;chr&gt; | data_category &lt;chr&gt; | data_format &lt;chr&gt; | data_type &lt;chr&gt; | platform &lt;chr&gt; | is_ffpe &lt;chr&gt; | passage &lt;chr&gt; | body_location &lt;chr&gt; | disease_type &lt;chr&gt; | comment &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 921 | WISTAR | wm3907n_R1.fastq.gz | WM3907 | wm3907n | NA | NA | wm3907n | Tumor | WES | ⋯ | Open Access | Raw Sequencing Data | fastq.gz | Raw Reads | Illumina | FALSE | NA | Skin | Melanoma | NA |\n",
       "\n",
       "\n",
       "$WUSTL\n",
       ":   \n",
       "A tibble: 1 × 26\n",
       "\n",
       "| ...1 &lt;dbl&gt; | contributor &lt;chr&gt; | file_name &lt;chr&gt; | patient_id &lt;chr&gt; | model_id &lt;chr&gt; | tumor_id &lt;chr&gt; | case_id &lt;chr&gt; | sample_id &lt;chr&gt; | sample_type &lt;chr&gt; | experimental_strategy &lt;chr&gt; | ⋯ ⋯ | access_level &lt;chr&gt; | data_category &lt;chr&gt; | data_format &lt;chr&gt; | data_type &lt;chr&gt; | platform &lt;chr&gt; | is_ffpe &lt;chr&gt; | passage &lt;chr&gt; | body_location &lt;chr&gt; | disease_type &lt;chr&gt; | comment &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1303 | WUSTL | 172845-121-T.r1.fastq.gz | PDMR-172845 | 172845-121-T_SL | 121-T | PDMR-172845 | 172845-121-T_SL | PDX | RNA-Seq | ⋯ | Open | Raw Sequencing Data | fastq.gz | Raw Reads | NovaSeq S4 | Unknown | P3 | Digestive/Gastrointestinal | Adenocarcinoma - colon | NA |\n",
       "\n",
       "\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "$BCM\n",
       "\u001b[90m# A tibble: 1 × 26\u001b[39m\n",
       "   ...1 contributor file_name     patient_id model_id tumor_id case_id sample_id\n",
       "  \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m         \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m      \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m    \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m    \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m   \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m    \n",
       "\u001b[90m1\u001b[39m     1 BCM         LZML_37246_N… 24452      BCM-0002 2011012… \u001b[31mNA\u001b[39m      BCM-0002 \n",
       "\u001b[90m# … with 18 more variables: sample_type <chr>, experimental_strategy <chr>,\u001b[39m\n",
       "\u001b[90m#   gender <chr>, paired_end <dbl>, capture_kit <chr>, capture_assembly <chr>,\u001b[39m\n",
       "\u001b[90m#   availability <chr>, public <chr>, access_level <chr>, data_category <chr>,\u001b[39m\n",
       "\u001b[90m#   data_format <chr>, data_type <chr>, platform <chr>, is_ffpe <chr>,\u001b[39m\n",
       "\u001b[90m#   passage <chr>, body_location <chr>, disease_type <chr>, comment <chr>\u001b[39m\n",
       "\n",
       "$HCI\n",
       "\u001b[90m# A tibble: 1 × 26\u001b[39m\n",
       "   ...1 contributor file_name     patient_id model_id tumor_id case_id sample_id\n",
       "  \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m         \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m      \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m    \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m    \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m   \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m    \n",
       "\u001b[90m1\u001b[39m   209 HCI         14990X12_HCI… HCI001     \u001b[31mNA\u001b[39m       \u001b[31mNA\u001b[39m       HCI001  14990X12 \n",
       "\u001b[90m# … with 18 more variables: sample_type <chr>, experimental_strategy <chr>,\u001b[39m\n",
       "\u001b[90m#   gender <chr>, paired_end <dbl>, capture_kit <chr>, capture_assembly <chr>,\u001b[39m\n",
       "\u001b[90m#   availability <chr>, public <chr>, access_level <chr>, data_category <chr>,\u001b[39m\n",
       "\u001b[90m#   data_format <chr>, data_type <chr>, platform <chr>, is_ffpe <chr>,\u001b[39m\n",
       "\u001b[90m#   passage <chr>, body_location <chr>, disease_type <chr>, comment <chr>\u001b[39m\n",
       "\n",
       "$MDACC\n",
       "\u001b[90m# A tibble: 1 × 26\u001b[39m\n",
       "   ...1 contributor file_name     patient_id model_id tumor_id case_id sample_id\n",
       "  \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m         \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m      \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m    \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m    \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m   \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m    \n",
       "\u001b[90m1\u001b[39m   455 MDACC       BFang-LungPD… TC211      \u001b[31mNA\u001b[39m       \u001b[31mNA\u001b[39m       TC211   TC211W   \n",
       "\u001b[90m# … with 18 more variables: sample_type <chr>, experimental_strategy <chr>,\u001b[39m\n",
       "\u001b[90m#   gender <chr>, paired_end <dbl>, capture_kit <chr>, capture_assembly <chr>,\u001b[39m\n",
       "\u001b[90m#   availability <chr>, public <chr>, access_level <chr>, data_category <chr>,\u001b[39m\n",
       "\u001b[90m#   data_format <chr>, data_type <chr>, platform <chr>, is_ffpe <chr>,\u001b[39m\n",
       "\u001b[90m#   passage <chr>, body_location <chr>, disease_type <chr>, comment <chr>\u001b[39m\n",
       "\n",
       "$`UC Davis`\n",
       "\u001b[90m# A tibble: 1 × 26\u001b[39m\n",
       "   ...1 contributor file_name     patient_id model_id tumor_id case_id sample_id\n",
       "  \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m         \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m      \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m    \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m    \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m   \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m    \n",
       "\u001b[90m1\u001b[39m   873 UC Davis    BL0269F404_P… BL0269F    BL0269F  BL0269F  BL0269F BL0269F  \n",
       "\u001b[90m# … with 18 more variables: sample_type <chr>, experimental_strategy <chr>,\u001b[39m\n",
       "\u001b[90m#   gender <chr>, paired_end <dbl>, capture_kit <chr>, capture_assembly <chr>,\u001b[39m\n",
       "\u001b[90m#   availability <chr>, public <chr>, access_level <chr>, data_category <chr>,\u001b[39m\n",
       "\u001b[90m#   data_format <chr>, data_type <chr>, platform <chr>, is_ffpe <chr>,\u001b[39m\n",
       "\u001b[90m#   passage <chr>, body_location <chr>, disease_type <chr>, comment <chr>\u001b[39m\n",
       "\n",
       "$WISTAR\n",
       "\u001b[90m# A tibble: 1 × 26\u001b[39m\n",
       "   ...1 contributor file_name     patient_id model_id tumor_id case_id sample_id\n",
       "  \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m         \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m      \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m    \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m    \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m   \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m    \n",
       "\u001b[90m1\u001b[39m   921 WISTAR      wm3907n_R1.f… WM3907     wm3907n  \u001b[31mNA\u001b[39m       \u001b[31mNA\u001b[39m      wm3907n  \n",
       "\u001b[90m# … with 18 more variables: sample_type <chr>, experimental_strategy <chr>,\u001b[39m\n",
       "\u001b[90m#   gender <chr>, paired_end <dbl>, capture_kit <chr>, capture_assembly <chr>,\u001b[39m\n",
       "\u001b[90m#   availability <chr>, public <chr>, access_level <chr>, data_category <chr>,\u001b[39m\n",
       "\u001b[90m#   data_format <chr>, data_type <chr>, platform <chr>, is_ffpe <chr>,\u001b[39m\n",
       "\u001b[90m#   passage <chr>, body_location <chr>, disease_type <chr>, comment <chr>\u001b[39m\n",
       "\n",
       "$WUSTL\n",
       "\u001b[90m# A tibble: 1 × 26\u001b[39m\n",
       "   ...1 contributor file_name   patient_id  model_id  tumor_id case_id sample_id\n",
       "  \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m     \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m    \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m   \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m    \n",
       "\u001b[90m1\u001b[39m  \u001b[4m1\u001b[24m303 WUSTL       172845-121… PDMR-172845 172845-1… 121-T    PDMR-1… 172845-1…\n",
       "\u001b[90m# … with 18 more variables: sample_type <chr>, experimental_strategy <chr>,\u001b[39m\n",
       "\u001b[90m#   gender <chr>, paired_end <dbl>, capture_kit <chr>, capture_assembly <chr>,\u001b[39m\n",
       "\u001b[90m#   availability <chr>, public <chr>, access_level <chr>, data_category <chr>,\u001b[39m\n",
       "\u001b[90m#   data_format <chr>, data_type <chr>, platform <chr>, is_ffpe <chr>,\u001b[39m\n",
       "\u001b[90m#   passage <chr>, body_location <chr>, disease_type <chr>, comment <chr>\u001b[39m\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Break down by PDTC\n",
    "\n",
    "pdtc_metadata <- pdtc_metadata %>% group_split(contributor)\n",
    "names(pdtc_metadata) <- c('BCM', 'HCI', 'MDACC', 'UC Davis', 'WISTAR', 'WUSTL')\n",
    "\n",
    "pdtc_metadata %>% map(head, 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "99ad697c-5e41-4fcc-8eb4-ab971a4fdb7d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<dl>\n",
       "\t<dt>$BCM</dt>\n",
       "\t\t<dd><table class=\"dataframe\">\n",
       "<caption>A data.frame: 26 × 1</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>.</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>...1</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>contributor</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>file_name</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>patient_id</th><td>  2</td></tr>\n",
       "\t<tr><th scope=row>model_id</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>tumor_id</th><td> 16</td></tr>\n",
       "\t<tr><th scope=row>case_id</th><td>208</td></tr>\n",
       "\t<tr><th scope=row>sample_id</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>sample_type</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>experimental_strategy</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>gender</th><td> 20</td></tr>\n",
       "\t<tr><th scope=row>paired_end</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>capture_kit</th><td> 26</td></tr>\n",
       "\t<tr><th scope=row>capture_assembly</th><td>208</td></tr>\n",
       "\t<tr><th scope=row>availability</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>public</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>access_level</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>data_category</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>data_format</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>data_type</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>platform</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>is_ffpe</th><td>124</td></tr>\n",
       "\t<tr><th scope=row>passage</th><td>208</td></tr>\n",
       "\t<tr><th scope=row>body_location</th><td> 48</td></tr>\n",
       "\t<tr><th scope=row>disease_type</th><td> 48</td></tr>\n",
       "\t<tr><th scope=row>comment</th><td>208</td></tr>\n",
       "</tbody>\n",
       "</table>\n",
       "</dd>\n",
       "\t<dt>$HCI</dt>\n",
       "\t\t<dd><table class=\"dataframe\">\n",
       "<caption>A data.frame: 26 × 1</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>.</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>...1</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>contributor</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>file_name</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>patient_id</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>model_id</th><td>246</td></tr>\n",
       "\t<tr><th scope=row>tumor_id</th><td>246</td></tr>\n",
       "\t<tr><th scope=row>case_id</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>sample_id</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>sample_type</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>experimental_strategy</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>gender</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>paired_end</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>capture_kit</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>capture_assembly</th><td> 48</td></tr>\n",
       "\t<tr><th scope=row>availability</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>public</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>access_level</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>data_category</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>data_format</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>data_type</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>platform</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>is_ffpe</th><td>246</td></tr>\n",
       "\t<tr><th scope=row>passage</th><td>144</td></tr>\n",
       "\t<tr><th scope=row>body_location</th><td>  8</td></tr>\n",
       "\t<tr><th scope=row>disease_type</th><td>  8</td></tr>\n",
       "\t<tr><th scope=row>comment</th><td>224</td></tr>\n",
       "</tbody>\n",
       "</table>\n",
       "</dd>\n",
       "\t<dt>$MDACC</dt>\n",
       "\t\t<dd><table class=\"dataframe\">\n",
       "<caption>A data.frame: 26 × 1</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>.</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>...1</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>contributor</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>file_name</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>patient_id</th><td> 12</td></tr>\n",
       "\t<tr><th scope=row>model_id</th><td>264</td></tr>\n",
       "\t<tr><th scope=row>tumor_id</th><td>418</td></tr>\n",
       "\t<tr><th scope=row>case_id</th><td> 12</td></tr>\n",
       "\t<tr><th scope=row>sample_id</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>sample_type</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>experimental_strategy</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>gender</th><td> 12</td></tr>\n",
       "\t<tr><th scope=row>paired_end</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>capture_kit</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>capture_assembly</th><td> 12</td></tr>\n",
       "\t<tr><th scope=row>availability</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>public</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>access_level</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>data_category</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>data_format</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>data_type</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>platform</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>is_ffpe</th><td>418</td></tr>\n",
       "\t<tr><th scope=row>passage</th><td>272</td></tr>\n",
       "\t<tr><th scope=row>body_location</th><td> 12</td></tr>\n",
       "\t<tr><th scope=row>disease_type</th><td> 12</td></tr>\n",
       "\t<tr><th scope=row>comment</th><td>404</td></tr>\n",
       "</tbody>\n",
       "</table>\n",
       "</dd>\n",
       "\t<dt>$`UC Davis`</dt>\n",
       "\t\t<dd><table class=\"dataframe\">\n",
       "<caption>A data.frame: 26 × 1</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>.</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>...1</th><td> 0</td></tr>\n",
       "\t<tr><th scope=row>contributor</th><td> 0</td></tr>\n",
       "\t<tr><th scope=row>file_name</th><td> 0</td></tr>\n",
       "\t<tr><th scope=row>patient_id</th><td> 0</td></tr>\n",
       "\t<tr><th scope=row>model_id</th><td> 0</td></tr>\n",
       "\t<tr><th scope=row>tumor_id</th><td> 0</td></tr>\n",
       "\t<tr><th scope=row>case_id</th><td> 0</td></tr>\n",
       "\t<tr><th scope=row>sample_id</th><td> 0</td></tr>\n",
       "\t<tr><th scope=row>sample_type</th><td> 0</td></tr>\n",
       "\t<tr><th scope=row>experimental_strategy</th><td> 0</td></tr>\n",
       "\t<tr><th scope=row>gender</th><td> 0</td></tr>\n",
       "\t<tr><th scope=row>paired_end</th><td> 0</td></tr>\n",
       "\t<tr><th scope=row>capture_kit</th><td> 0</td></tr>\n",
       "\t<tr><th scope=row>capture_assembly</th><td>48</td></tr>\n",
       "\t<tr><th scope=row>availability</th><td> 0</td></tr>\n",
       "\t<tr><th scope=row>public</th><td> 0</td></tr>\n",
       "\t<tr><th scope=row>access_level</th><td> 0</td></tr>\n",
       "\t<tr><th scope=row>data_category</th><td> 0</td></tr>\n",
       "\t<tr><th scope=row>data_format</th><td> 0</td></tr>\n",
       "\t<tr><th scope=row>data_type</th><td> 0</td></tr>\n",
       "\t<tr><th scope=row>platform</th><td> 0</td></tr>\n",
       "\t<tr><th scope=row>is_ffpe</th><td>48</td></tr>\n",
       "\t<tr><th scope=row>passage</th><td> 0</td></tr>\n",
       "\t<tr><th scope=row>body_location</th><td> 0</td></tr>\n",
       "\t<tr><th scope=row>disease_type</th><td> 0</td></tr>\n",
       "\t<tr><th scope=row>comment</th><td>48</td></tr>\n",
       "</tbody>\n",
       "</table>\n",
       "</dd>\n",
       "\t<dt>$WISTAR</dt>\n",
       "\t\t<dd><table class=\"dataframe\">\n",
       "<caption>A data.frame: 26 × 1</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>.</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>...1</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>contributor</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>file_name</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>patient_id</th><td> 12</td></tr>\n",
       "\t<tr><th scope=row>model_id</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>tumor_id</th><td>382</td></tr>\n",
       "\t<tr><th scope=row>case_id</th><td>382</td></tr>\n",
       "\t<tr><th scope=row>sample_id</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>sample_type</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>experimental_strategy</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>gender</th><td> 16</td></tr>\n",
       "\t<tr><th scope=row>paired_end</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>capture_kit</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>capture_assembly</th><td>382</td></tr>\n",
       "\t<tr><th scope=row>availability</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>public</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>access_level</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>data_category</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>data_format</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>data_type</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>platform</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>is_ffpe</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>passage</th><td>190</td></tr>\n",
       "\t<tr><th scope=row>body_location</th><td> 12</td></tr>\n",
       "\t<tr><th scope=row>disease_type</th><td> 12</td></tr>\n",
       "\t<tr><th scope=row>comment</th><td>382</td></tr>\n",
       "</tbody>\n",
       "</table>\n",
       "</dd>\n",
       "\t<dt>$WUSTL</dt>\n",
       "\t\t<dd><table class=\"dataframe\">\n",
       "<caption>A data.frame: 26 × 1</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>.</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>...1</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>contributor</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>file_name</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>patient_id</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>model_id</th><td>360</td></tr>\n",
       "\t<tr><th scope=row>tumor_id</th><td>360</td></tr>\n",
       "\t<tr><th scope=row>case_id</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>sample_id</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>sample_type</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>experimental_strategy</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>gender</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>paired_end</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>capture_kit</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>capture_assembly</th><td>872</td></tr>\n",
       "\t<tr><th scope=row>availability</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>public</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>access_level</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>data_category</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>data_format</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>data_type</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>platform</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>is_ffpe</th><td>878</td></tr>\n",
       "\t<tr><th scope=row>passage</th><td>360</td></tr>\n",
       "\t<tr><th scope=row>body_location</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>disease_type</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>comment</th><td>878</td></tr>\n",
       "</tbody>\n",
       "</table>\n",
       "</dd>\n",
       "</dl>\n"
      ],
      "text/latex": [
       "\\begin{description}\n",
       "\\item[\\$BCM] A data.frame: 26 × 1\n",
       "\\begin{tabular}{r|l}\n",
       "  & .\\\\\n",
       "  & <dbl>\\\\\n",
       "\\hline\n",
       "\t...1 &   0\\\\\n",
       "\tcontributor &   0\\\\\n",
       "\tfile\\_name &   0\\\\\n",
       "\tpatient\\_id &   2\\\\\n",
       "\tmodel\\_id &   0\\\\\n",
       "\ttumor\\_id &  16\\\\\n",
       "\tcase\\_id & 208\\\\\n",
       "\tsample\\_id &   0\\\\\n",
       "\tsample\\_type &   0\\\\\n",
       "\texperimental\\_strategy &   0\\\\\n",
       "\tgender &  20\\\\\n",
       "\tpaired\\_end &   0\\\\\n",
       "\tcapture\\_kit &  26\\\\\n",
       "\tcapture\\_assembly & 208\\\\\n",
       "\tavailability &   0\\\\\n",
       "\tpublic &   0\\\\\n",
       "\taccess\\_level &   0\\\\\n",
       "\tdata\\_category &   0\\\\\n",
       "\tdata\\_format &   0\\\\\n",
       "\tdata\\_type &   0\\\\\n",
       "\tplatform &   0\\\\\n",
       "\tis\\_ffpe & 124\\\\\n",
       "\tpassage & 208\\\\\n",
       "\tbody\\_location &  48\\\\\n",
       "\tdisease\\_type &  48\\\\\n",
       "\tcomment & 208\\\\\n",
       "\\end{tabular}\n",
       "\n",
       "\\item[\\$HCI] A data.frame: 26 × 1\n",
       "\\begin{tabular}{r|l}\n",
       "  & .\\\\\n",
       "  & <dbl>\\\\\n",
       "\\hline\n",
       "\t...1 &   0\\\\\n",
       "\tcontributor &   0\\\\\n",
       "\tfile\\_name &   0\\\\\n",
       "\tpatient\\_id &   0\\\\\n",
       "\tmodel\\_id & 246\\\\\n",
       "\ttumor\\_id & 246\\\\\n",
       "\tcase\\_id &   0\\\\\n",
       "\tsample\\_id &   0\\\\\n",
       "\tsample\\_type &   0\\\\\n",
       "\texperimental\\_strategy &   0\\\\\n",
       "\tgender &   0\\\\\n",
       "\tpaired\\_end &   0\\\\\n",
       "\tcapture\\_kit &   0\\\\\n",
       "\tcapture\\_assembly &  48\\\\\n",
       "\tavailability &   0\\\\\n",
       "\tpublic &   0\\\\\n",
       "\taccess\\_level &   0\\\\\n",
       "\tdata\\_category &   0\\\\\n",
       "\tdata\\_format &   0\\\\\n",
       "\tdata\\_type &   0\\\\\n",
       "\tplatform &   0\\\\\n",
       "\tis\\_ffpe & 246\\\\\n",
       "\tpassage & 144\\\\\n",
       "\tbody\\_location &   8\\\\\n",
       "\tdisease\\_type &   8\\\\\n",
       "\tcomment & 224\\\\\n",
       "\\end{tabular}\n",
       "\n",
       "\\item[\\$MDACC] A data.frame: 26 × 1\n",
       "\\begin{tabular}{r|l}\n",
       "  & .\\\\\n",
       "  & <dbl>\\\\\n",
       "\\hline\n",
       "\t...1 &   0\\\\\n",
       "\tcontributor &   0\\\\\n",
       "\tfile\\_name &   0\\\\\n",
       "\tpatient\\_id &  12\\\\\n",
       "\tmodel\\_id & 264\\\\\n",
       "\ttumor\\_id & 418\\\\\n",
       "\tcase\\_id &  12\\\\\n",
       "\tsample\\_id &   0\\\\\n",
       "\tsample\\_type &   0\\\\\n",
       "\texperimental\\_strategy &   0\\\\\n",
       "\tgender &  12\\\\\n",
       "\tpaired\\_end &   0\\\\\n",
       "\tcapture\\_kit &   0\\\\\n",
       "\tcapture\\_assembly &  12\\\\\n",
       "\tavailability &   0\\\\\n",
       "\tpublic &   0\\\\\n",
       "\taccess\\_level &   0\\\\\n",
       "\tdata\\_category &   0\\\\\n",
       "\tdata\\_format &   0\\\\\n",
       "\tdata\\_type &   0\\\\\n",
       "\tplatform &   0\\\\\n",
       "\tis\\_ffpe & 418\\\\\n",
       "\tpassage & 272\\\\\n",
       "\tbody\\_location &  12\\\\\n",
       "\tdisease\\_type &  12\\\\\n",
       "\tcomment & 404\\\\\n",
       "\\end{tabular}\n",
       "\n",
       "\\item[\\$`UC Davis`] A data.frame: 26 × 1\n",
       "\\begin{tabular}{r|l}\n",
       "  & .\\\\\n",
       "  & <dbl>\\\\\n",
       "\\hline\n",
       "\t...1 &  0\\\\\n",
       "\tcontributor &  0\\\\\n",
       "\tfile\\_name &  0\\\\\n",
       "\tpatient\\_id &  0\\\\\n",
       "\tmodel\\_id &  0\\\\\n",
       "\ttumor\\_id &  0\\\\\n",
       "\tcase\\_id &  0\\\\\n",
       "\tsample\\_id &  0\\\\\n",
       "\tsample\\_type &  0\\\\\n",
       "\texperimental\\_strategy &  0\\\\\n",
       "\tgender &  0\\\\\n",
       "\tpaired\\_end &  0\\\\\n",
       "\tcapture\\_kit &  0\\\\\n",
       "\tcapture\\_assembly & 48\\\\\n",
       "\tavailability &  0\\\\\n",
       "\tpublic &  0\\\\\n",
       "\taccess\\_level &  0\\\\\n",
       "\tdata\\_category &  0\\\\\n",
       "\tdata\\_format &  0\\\\\n",
       "\tdata\\_type &  0\\\\\n",
       "\tplatform &  0\\\\\n",
       "\tis\\_ffpe & 48\\\\\n",
       "\tpassage &  0\\\\\n",
       "\tbody\\_location &  0\\\\\n",
       "\tdisease\\_type &  0\\\\\n",
       "\tcomment & 48\\\\\n",
       "\\end{tabular}\n",
       "\n",
       "\\item[\\$WISTAR] A data.frame: 26 × 1\n",
       "\\begin{tabular}{r|l}\n",
       "  & .\\\\\n",
       "  & <dbl>\\\\\n",
       "\\hline\n",
       "\t...1 &   0\\\\\n",
       "\tcontributor &   0\\\\\n",
       "\tfile\\_name &   0\\\\\n",
       "\tpatient\\_id &  12\\\\\n",
       "\tmodel\\_id &   0\\\\\n",
       "\ttumor\\_id & 382\\\\\n",
       "\tcase\\_id & 382\\\\\n",
       "\tsample\\_id &   0\\\\\n",
       "\tsample\\_type &   0\\\\\n",
       "\texperimental\\_strategy &   0\\\\\n",
       "\tgender &  16\\\\\n",
       "\tpaired\\_end &   0\\\\\n",
       "\tcapture\\_kit &   0\\\\\n",
       "\tcapture\\_assembly & 382\\\\\n",
       "\tavailability &   0\\\\\n",
       "\tpublic &   0\\\\\n",
       "\taccess\\_level &   0\\\\\n",
       "\tdata\\_category &   0\\\\\n",
       "\tdata\\_format &   0\\\\\n",
       "\tdata\\_type &   0\\\\\n",
       "\tplatform &   0\\\\\n",
       "\tis\\_ffpe &   0\\\\\n",
       "\tpassage & 190\\\\\n",
       "\tbody\\_location &  12\\\\\n",
       "\tdisease\\_type &  12\\\\\n",
       "\tcomment & 382\\\\\n",
       "\\end{tabular}\n",
       "\n",
       "\\item[\\$WUSTL] A data.frame: 26 × 1\n",
       "\\begin{tabular}{r|l}\n",
       "  & .\\\\\n",
       "  & <dbl>\\\\\n",
       "\\hline\n",
       "\t...1 &   0\\\\\n",
       "\tcontributor &   0\\\\\n",
       "\tfile\\_name &   0\\\\\n",
       "\tpatient\\_id &   0\\\\\n",
       "\tmodel\\_id & 360\\\\\n",
       "\ttumor\\_id & 360\\\\\n",
       "\tcase\\_id &   0\\\\\n",
       "\tsample\\_id &   0\\\\\n",
       "\tsample\\_type &   0\\\\\n",
       "\texperimental\\_strategy &   0\\\\\n",
       "\tgender &   0\\\\\n",
       "\tpaired\\_end &   0\\\\\n",
       "\tcapture\\_kit &   0\\\\\n",
       "\tcapture\\_assembly & 872\\\\\n",
       "\tavailability &   0\\\\\n",
       "\tpublic &   0\\\\\n",
       "\taccess\\_level &   0\\\\\n",
       "\tdata\\_category &   0\\\\\n",
       "\tdata\\_format &   0\\\\\n",
       "\tdata\\_type &   0\\\\\n",
       "\tplatform &   0\\\\\n",
       "\tis\\_ffpe & 878\\\\\n",
       "\tpassage & 360\\\\\n",
       "\tbody\\_location &   0\\\\\n",
       "\tdisease\\_type &   0\\\\\n",
       "\tcomment & 878\\\\\n",
       "\\end{tabular}\n",
       "\n",
       "\\end{description}\n"
      ],
      "text/markdown": [
       "$BCM\n",
       ":   \n",
       "A data.frame: 26 × 1\n",
       "\n",
       "| <!--/--> | . &lt;dbl&gt; |\n",
       "|---|---|\n",
       "| ...1 |   0 |\n",
       "| contributor |   0 |\n",
       "| file_name |   0 |\n",
       "| patient_id |   2 |\n",
       "| model_id |   0 |\n",
       "| tumor_id |  16 |\n",
       "| case_id | 208 |\n",
       "| sample_id |   0 |\n",
       "| sample_type |   0 |\n",
       "| experimental_strategy |   0 |\n",
       "| gender |  20 |\n",
       "| paired_end |   0 |\n",
       "| capture_kit |  26 |\n",
       "| capture_assembly | 208 |\n",
       "| availability |   0 |\n",
       "| public |   0 |\n",
       "| access_level |   0 |\n",
       "| data_category |   0 |\n",
       "| data_format |   0 |\n",
       "| data_type |   0 |\n",
       "| platform |   0 |\n",
       "| is_ffpe | 124 |\n",
       "| passage | 208 |\n",
       "| body_location |  48 |\n",
       "| disease_type |  48 |\n",
       "| comment | 208 |\n",
       "\n",
       "\n",
       "$HCI\n",
       ":   \n",
       "A data.frame: 26 × 1\n",
       "\n",
       "| <!--/--> | . &lt;dbl&gt; |\n",
       "|---|---|\n",
       "| ...1 |   0 |\n",
       "| contributor |   0 |\n",
       "| file_name |   0 |\n",
       "| patient_id |   0 |\n",
       "| model_id | 246 |\n",
       "| tumor_id | 246 |\n",
       "| case_id |   0 |\n",
       "| sample_id |   0 |\n",
       "| sample_type |   0 |\n",
       "| experimental_strategy |   0 |\n",
       "| gender |   0 |\n",
       "| paired_end |   0 |\n",
       "| capture_kit |   0 |\n",
       "| capture_assembly |  48 |\n",
       "| availability |   0 |\n",
       "| public |   0 |\n",
       "| access_level |   0 |\n",
       "| data_category |   0 |\n",
       "| data_format |   0 |\n",
       "| data_type |   0 |\n",
       "| platform |   0 |\n",
       "| is_ffpe | 246 |\n",
       "| passage | 144 |\n",
       "| body_location |   8 |\n",
       "| disease_type |   8 |\n",
       "| comment | 224 |\n",
       "\n",
       "\n",
       "$MDACC\n",
       ":   \n",
       "A data.frame: 26 × 1\n",
       "\n",
       "| <!--/--> | . &lt;dbl&gt; |\n",
       "|---|---|\n",
       "| ...1 |   0 |\n",
       "| contributor |   0 |\n",
       "| file_name |   0 |\n",
       "| patient_id |  12 |\n",
       "| model_id | 264 |\n",
       "| tumor_id | 418 |\n",
       "| case_id |  12 |\n",
       "| sample_id |   0 |\n",
       "| sample_type |   0 |\n",
       "| experimental_strategy |   0 |\n",
       "| gender |  12 |\n",
       "| paired_end |   0 |\n",
       "| capture_kit |   0 |\n",
       "| capture_assembly |  12 |\n",
       "| availability |   0 |\n",
       "| public |   0 |\n",
       "| access_level |   0 |\n",
       "| data_category |   0 |\n",
       "| data_format |   0 |\n",
       "| data_type |   0 |\n",
       "| platform |   0 |\n",
       "| is_ffpe | 418 |\n",
       "| passage | 272 |\n",
       "| body_location |  12 |\n",
       "| disease_type |  12 |\n",
       "| comment | 404 |\n",
       "\n",
       "\n",
       "$`UC Davis`\n",
       ":   \n",
       "A data.frame: 26 × 1\n",
       "\n",
       "| <!--/--> | . &lt;dbl&gt; |\n",
       "|---|---|\n",
       "| ...1 |  0 |\n",
       "| contributor |  0 |\n",
       "| file_name |  0 |\n",
       "| patient_id |  0 |\n",
       "| model_id |  0 |\n",
       "| tumor_id |  0 |\n",
       "| case_id |  0 |\n",
       "| sample_id |  0 |\n",
       "| sample_type |  0 |\n",
       "| experimental_strategy |  0 |\n",
       "| gender |  0 |\n",
       "| paired_end |  0 |\n",
       "| capture_kit |  0 |\n",
       "| capture_assembly | 48 |\n",
       "| availability |  0 |\n",
       "| public |  0 |\n",
       "| access_level |  0 |\n",
       "| data_category |  0 |\n",
       "| data_format |  0 |\n",
       "| data_type |  0 |\n",
       "| platform |  0 |\n",
       "| is_ffpe | 48 |\n",
       "| passage |  0 |\n",
       "| body_location |  0 |\n",
       "| disease_type |  0 |\n",
       "| comment | 48 |\n",
       "\n",
       "\n",
       "$WISTAR\n",
       ":   \n",
       "A data.frame: 26 × 1\n",
       "\n",
       "| <!--/--> | . &lt;dbl&gt; |\n",
       "|---|---|\n",
       "| ...1 |   0 |\n",
       "| contributor |   0 |\n",
       "| file_name |   0 |\n",
       "| patient_id |  12 |\n",
       "| model_id |   0 |\n",
       "| tumor_id | 382 |\n",
       "| case_id | 382 |\n",
       "| sample_id |   0 |\n",
       "| sample_type |   0 |\n",
       "| experimental_strategy |   0 |\n",
       "| gender |  16 |\n",
       "| paired_end |   0 |\n",
       "| capture_kit |   0 |\n",
       "| capture_assembly | 382 |\n",
       "| availability |   0 |\n",
       "| public |   0 |\n",
       "| access_level |   0 |\n",
       "| data_category |   0 |\n",
       "| data_format |   0 |\n",
       "| data_type |   0 |\n",
       "| platform |   0 |\n",
       "| is_ffpe |   0 |\n",
       "| passage | 190 |\n",
       "| body_location |  12 |\n",
       "| disease_type |  12 |\n",
       "| comment | 382 |\n",
       "\n",
       "\n",
       "$WUSTL\n",
       ":   \n",
       "A data.frame: 26 × 1\n",
       "\n",
       "| <!--/--> | . &lt;dbl&gt; |\n",
       "|---|---|\n",
       "| ...1 |   0 |\n",
       "| contributor |   0 |\n",
       "| file_name |   0 |\n",
       "| patient_id |   0 |\n",
       "| model_id | 360 |\n",
       "| tumor_id | 360 |\n",
       "| case_id |   0 |\n",
       "| sample_id |   0 |\n",
       "| sample_type |   0 |\n",
       "| experimental_strategy |   0 |\n",
       "| gender |   0 |\n",
       "| paired_end |   0 |\n",
       "| capture_kit |   0 |\n",
       "| capture_assembly | 872 |\n",
       "| availability |   0 |\n",
       "| public |   0 |\n",
       "| access_level |   0 |\n",
       "| data_category |   0 |\n",
       "| data_format |   0 |\n",
       "| data_type |   0 |\n",
       "| platform |   0 |\n",
       "| is_ffpe | 878 |\n",
       "| passage | 360 |\n",
       "| body_location |   0 |\n",
       "| disease_type |   0 |\n",
       "| comment | 878 |\n",
       "\n",
       "\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "$BCM\n",
       "                        .\n",
       "...1                    0\n",
       "contributor             0\n",
       "file_name               0\n",
       "patient_id              2\n",
       "model_id                0\n",
       "tumor_id               16\n",
       "case_id               208\n",
       "sample_id               0\n",
       "sample_type             0\n",
       "experimental_strategy   0\n",
       "gender                 20\n",
       "paired_end              0\n",
       "capture_kit            26\n",
       "capture_assembly      208\n",
       "availability            0\n",
       "public                  0\n",
       "access_level            0\n",
       "data_category           0\n",
       "data_format             0\n",
       "data_type               0\n",
       "platform                0\n",
       "is_ffpe               124\n",
       "passage               208\n",
       "body_location          48\n",
       "disease_type           48\n",
       "comment               208\n",
       "\n",
       "$HCI\n",
       "                        .\n",
       "...1                    0\n",
       "contributor             0\n",
       "file_name               0\n",
       "patient_id              0\n",
       "model_id              246\n",
       "tumor_id              246\n",
       "case_id                 0\n",
       "sample_id               0\n",
       "sample_type             0\n",
       "experimental_strategy   0\n",
       "gender                  0\n",
       "paired_end              0\n",
       "capture_kit             0\n",
       "capture_assembly       48\n",
       "availability            0\n",
       "public                  0\n",
       "access_level            0\n",
       "data_category           0\n",
       "data_format             0\n",
       "data_type               0\n",
       "platform                0\n",
       "is_ffpe               246\n",
       "passage               144\n",
       "body_location           8\n",
       "disease_type            8\n",
       "comment               224\n",
       "\n",
       "$MDACC\n",
       "                        .\n",
       "...1                    0\n",
       "contributor             0\n",
       "file_name               0\n",
       "patient_id             12\n",
       "model_id              264\n",
       "tumor_id              418\n",
       "case_id                12\n",
       "sample_id               0\n",
       "sample_type             0\n",
       "experimental_strategy   0\n",
       "gender                 12\n",
       "paired_end              0\n",
       "capture_kit             0\n",
       "capture_assembly       12\n",
       "availability            0\n",
       "public                  0\n",
       "access_level            0\n",
       "data_category           0\n",
       "data_format             0\n",
       "data_type               0\n",
       "platform                0\n",
       "is_ffpe               418\n",
       "passage               272\n",
       "body_location          12\n",
       "disease_type           12\n",
       "comment               404\n",
       "\n",
       "$`UC Davis`\n",
       "                       .\n",
       "...1                   0\n",
       "contributor            0\n",
       "file_name              0\n",
       "patient_id             0\n",
       "model_id               0\n",
       "tumor_id               0\n",
       "case_id                0\n",
       "sample_id              0\n",
       "sample_type            0\n",
       "experimental_strategy  0\n",
       "gender                 0\n",
       "paired_end             0\n",
       "capture_kit            0\n",
       "capture_assembly      48\n",
       "availability           0\n",
       "public                 0\n",
       "access_level           0\n",
       "data_category          0\n",
       "data_format            0\n",
       "data_type              0\n",
       "platform               0\n",
       "is_ffpe               48\n",
       "passage                0\n",
       "body_location          0\n",
       "disease_type           0\n",
       "comment               48\n",
       "\n",
       "$WISTAR\n",
       "                        .\n",
       "...1                    0\n",
       "contributor             0\n",
       "file_name               0\n",
       "patient_id             12\n",
       "model_id                0\n",
       "tumor_id              382\n",
       "case_id               382\n",
       "sample_id               0\n",
       "sample_type             0\n",
       "experimental_strategy   0\n",
       "gender                 16\n",
       "paired_end              0\n",
       "capture_kit             0\n",
       "capture_assembly      382\n",
       "availability            0\n",
       "public                  0\n",
       "access_level            0\n",
       "data_category           0\n",
       "data_format             0\n",
       "data_type               0\n",
       "platform                0\n",
       "is_ffpe                 0\n",
       "passage               190\n",
       "body_location          12\n",
       "disease_type           12\n",
       "comment               382\n",
       "\n",
       "$WUSTL\n",
       "                        .\n",
       "...1                    0\n",
       "contributor             0\n",
       "file_name               0\n",
       "patient_id              0\n",
       "model_id              360\n",
       "tumor_id              360\n",
       "case_id                 0\n",
       "sample_id               0\n",
       "sample_type             0\n",
       "experimental_strategy   0\n",
       "gender                  0\n",
       "paired_end              0\n",
       "capture_kit             0\n",
       "capture_assembly      872\n",
       "availability            0\n",
       "public                  0\n",
       "access_level            0\n",
       "data_category           0\n",
       "data_format             0\n",
       "data_type               0\n",
       "platform                0\n",
       "is_ffpe               878\n",
       "passage               360\n",
       "body_location           0\n",
       "disease_type            0\n",
       "comment               878\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Break down the missing metadata fields by PDTC\n",
    "\n",
    "pdtc_metadata %>% map(summarize_missing_fields)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "43d8a09a-0843-4bf6-8eee-b5983b811c34",
   "metadata": {},
   "source": [
    "### How many missing metadata fields were there originally?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "a9436a42-2f8e-438e-b587-22043d24b1e9",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "New names:\n",
      "* `` -> ...1\n",
      "\n",
      "\u001b[1mRows: \u001b[22m\u001b[34m2822\u001b[39m \u001b[1mColumns: \u001b[22m\u001b[34m29\u001b[39m\n",
      "\n",
      "\u001b[36m──\u001b[39m \u001b[1mColumn specification\u001b[22m \u001b[36m──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────\u001b[39m\n",
      "\u001b[1mDelimiter:\u001b[22m \",\"\n",
      "\u001b[31mchr\u001b[39m (23): contributor, file_name, patient_id, model_id, tumor_id, case_id, s...\n",
      "\u001b[32mdbl\u001b[39m  (4): ...1, paired_end, created_datetime, file_size\n",
      "\u001b[33mlgl\u001b[39m  (2): is_ffpe, investigation\n",
      "\n",
      "\n",
      "\u001b[36mℹ\u001b[39m Use \u001b[30m\u001b[47m\u001b[30m\u001b[47m`spec()`\u001b[47m\u001b[30m\u001b[49m\u001b[39m to retrieve the full column specification for this data.\n",
      "\u001b[36mℹ\u001b[39m Specify the column types or set \u001b[30m\u001b[47m\u001b[30m\u001b[47m`show_col_types = FALSE`\u001b[47m\u001b[30m\u001b[49m\u001b[39m to quiet this message.\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "2822"
      ],
      "text/latex": [
       "2822"
      ],
      "text/markdown": [
       "2822"
      ],
      "text/plain": [
       "[1] 2822"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 29</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>...1</th><th scope=col>contributor</th><th scope=col>file_name</th><th scope=col>patient_id</th><th scope=col>model_id</th><th scope=col>tumor_id</th><th scope=col>case_id</th><th scope=col>sample_id</th><th scope=col>sample_type</th><th scope=col>experimental_strategy</th><th scope=col>⋯</th><th scope=col>data_format</th><th scope=col>data_type</th><th scope=col>platform</th><th scope=col>is_ffpe</th><th scope=col>file_size</th><th scope=col>passage</th><th scope=col>disease_type</th><th scope=col>comment</th><th scope=col>investigation</th><th scope=col>disease_detail</th></tr>\n",
       "\t<tr><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td>BCM</td><td>LZML_37246_NoIndex_L00X_R1_001.fastq.gz</td><td>24452</td><td>BCM-0002</td><td>2011012-2  </td><td>NA</td><td>BCM-0002</td><td>PDX</td><td>RNA-Seq</td><td>⋯</td><td>FASTQ.GZ</td><td>Raw Reads</td><td>Illumina</td><td>FALSE</td><td>20656964441</td><td>NA</td><td>Breast</td><td>NA</td><td>NA</td><td>Invasive breast carcinoma</td></tr>\n",
       "\t<tr><td>2</td><td>BCM</td><td>LZML_37246_NoIndex_L00X_R2_001.fastq.gz</td><td>24452</td><td>BCM-0002</td><td>2011012-2  </td><td>NA</td><td>BCM-0002</td><td>PDX</td><td>RNA-Seq</td><td>⋯</td><td>FASTQ.GZ</td><td>Raw Reads</td><td>Illumina</td><td>FALSE</td><td>21093643386</td><td>NA</td><td>Breast</td><td>NA</td><td>NA</td><td>Invasive breast carcinoma</td></tr>\n",
       "\t<tr><td>3</td><td>BCM</td><td>ML6065_GATCAGCG_L00X_R1_001.fastq.gz   </td><td>2219 </td><td>BCM-0046</td><td>2011012-46 </td><td>NA</td><td>BCM-0046</td><td>PDX</td><td>RNA-Seq</td><td>⋯</td><td>FASTQ.GZ</td><td>Raw Reads</td><td>Illumina</td><td>FALSE</td><td>22027304876</td><td>NA</td><td>Breast</td><td>NA</td><td>NA</td><td>Invasive breast carcinoma</td></tr>\n",
       "\t<tr><td>4</td><td>BCM</td><td>ML6065_GATCAGCG_L00X_R2_001.fastq.gz   </td><td>2219 </td><td>BCM-0046</td><td>2011012-46 </td><td>NA</td><td>BCM-0046</td><td>PDX</td><td>RNA-Seq</td><td>⋯</td><td>FASTQ.GZ</td><td>Raw Reads</td><td>Illumina</td><td>FALSE</td><td>22027304876</td><td>NA</td><td>Breast</td><td>NA</td><td>NA</td><td>Invasive breast carcinoma</td></tr>\n",
       "\t<tr><td>5</td><td>BCM</td><td>ML6061_CAGATCTG_L00X_R1_001.fastq.gz   </td><td>24561</td><td>BCM-0104</td><td>2011012-104</td><td>NA</td><td>BCM-0104</td><td>PDX</td><td>RNA-Seq</td><td>⋯</td><td>FASTQ.GZ</td><td>Raw Reads</td><td>Illumina</td><td>FALSE</td><td>17927987899</td><td>NA</td><td>Breast</td><td>NA</td><td>NA</td><td>Invasive breast carcinoma</td></tr>\n",
       "\t<tr><td>6</td><td>BCM</td><td>ML6061_CAGATCTG_L00X_R2_001.fastq.gz   </td><td>24561</td><td>BCM-0104</td><td>2011012-104</td><td>NA</td><td>BCM-0104</td><td>PDX</td><td>RNA-Seq</td><td>⋯</td><td>FASTQ.GZ</td><td>Raw Reads</td><td>Illumina</td><td>FALSE</td><td>18002109303</td><td>NA</td><td>Breast</td><td>NA</td><td>NA</td><td>Invasive breast carcinoma</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 29\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " ...1 & contributor & file\\_name & patient\\_id & model\\_id & tumor\\_id & case\\_id & sample\\_id & sample\\_type & experimental\\_strategy & ⋯ & data\\_format & data\\_type & platform & is\\_ffpe & file\\_size & passage & disease\\_type & comment & investigation & disease\\_detail\\\\\n",
       " <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & ⋯ & <chr> & <chr> & <chr> & <lgl> & <dbl> & <chr> & <chr> & <chr> & <lgl> & <chr>\\\\\n",
       "\\hline\n",
       "\t 1 & BCM & LZML\\_37246\\_NoIndex\\_L00X\\_R1\\_001.fastq.gz & 24452 & BCM-0002 & 2011012-2   & NA & BCM-0002 & PDX & RNA-Seq & ⋯ & FASTQ.GZ & Raw Reads & Illumina & FALSE & 20656964441 & NA & Breast & NA & NA & Invasive breast carcinoma\\\\\n",
       "\t 2 & BCM & LZML\\_37246\\_NoIndex\\_L00X\\_R2\\_001.fastq.gz & 24452 & BCM-0002 & 2011012-2   & NA & BCM-0002 & PDX & RNA-Seq & ⋯ & FASTQ.GZ & Raw Reads & Illumina & FALSE & 21093643386 & NA & Breast & NA & NA & Invasive breast carcinoma\\\\\n",
       "\t 3 & BCM & ML6065\\_GATCAGCG\\_L00X\\_R1\\_001.fastq.gz    & 2219  & BCM-0046 & 2011012-46  & NA & BCM-0046 & PDX & RNA-Seq & ⋯ & FASTQ.GZ & Raw Reads & Illumina & FALSE & 22027304876 & NA & Breast & NA & NA & Invasive breast carcinoma\\\\\n",
       "\t 4 & BCM & ML6065\\_GATCAGCG\\_L00X\\_R2\\_001.fastq.gz    & 2219  & BCM-0046 & 2011012-46  & NA & BCM-0046 & PDX & RNA-Seq & ⋯ & FASTQ.GZ & Raw Reads & Illumina & FALSE & 22027304876 & NA & Breast & NA & NA & Invasive breast carcinoma\\\\\n",
       "\t 5 & BCM & ML6061\\_CAGATCTG\\_L00X\\_R1\\_001.fastq.gz    & 24561 & BCM-0104 & 2011012-104 & NA & BCM-0104 & PDX & RNA-Seq & ⋯ & FASTQ.GZ & Raw Reads & Illumina & FALSE & 17927987899 & NA & Breast & NA & NA & Invasive breast carcinoma\\\\\n",
       "\t 6 & BCM & ML6061\\_CAGATCTG\\_L00X\\_R2\\_001.fastq.gz    & 24561 & BCM-0104 & 2011012-104 & NA & BCM-0104 & PDX & RNA-Seq & ⋯ & FASTQ.GZ & Raw Reads & Illumina & FALSE & 18002109303 & NA & Breast & NA & NA & Invasive breast carcinoma\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 29\n",
       "\n",
       "| ...1 &lt;dbl&gt; | contributor &lt;chr&gt; | file_name &lt;chr&gt; | patient_id &lt;chr&gt; | model_id &lt;chr&gt; | tumor_id &lt;chr&gt; | case_id &lt;chr&gt; | sample_id &lt;chr&gt; | sample_type &lt;chr&gt; | experimental_strategy &lt;chr&gt; | ⋯ ⋯ | data_format &lt;chr&gt; | data_type &lt;chr&gt; | platform &lt;chr&gt; | is_ffpe &lt;lgl&gt; | file_size &lt;dbl&gt; | passage &lt;chr&gt; | disease_type &lt;chr&gt; | comment &lt;chr&gt; | investigation &lt;lgl&gt; | disease_detail &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | BCM | LZML_37246_NoIndex_L00X_R1_001.fastq.gz | 24452 | BCM-0002 | 2011012-2   | NA | BCM-0002 | PDX | RNA-Seq | ⋯ | FASTQ.GZ | Raw Reads | Illumina | FALSE | 20656964441 | NA | Breast | NA | NA | Invasive breast carcinoma |\n",
       "| 2 | BCM | LZML_37246_NoIndex_L00X_R2_001.fastq.gz | 24452 | BCM-0002 | 2011012-2   | NA | BCM-0002 | PDX | RNA-Seq | ⋯ | FASTQ.GZ | Raw Reads | Illumina | FALSE | 21093643386 | NA | Breast | NA | NA | Invasive breast carcinoma |\n",
       "| 3 | BCM | ML6065_GATCAGCG_L00X_R1_001.fastq.gz    | 2219  | BCM-0046 | 2011012-46  | NA | BCM-0046 | PDX | RNA-Seq | ⋯ | FASTQ.GZ | Raw Reads | Illumina | FALSE | 22027304876 | NA | Breast | NA | NA | Invasive breast carcinoma |\n",
       "| 4 | BCM | ML6065_GATCAGCG_L00X_R2_001.fastq.gz    | 2219  | BCM-0046 | 2011012-46  | NA | BCM-0046 | PDX | RNA-Seq | ⋯ | FASTQ.GZ | Raw Reads | Illumina | FALSE | 22027304876 | NA | Breast | NA | NA | Invasive breast carcinoma |\n",
       "| 5 | BCM | ML6061_CAGATCTG_L00X_R1_001.fastq.gz    | 24561 | BCM-0104 | 2011012-104 | NA | BCM-0104 | PDX | RNA-Seq | ⋯ | FASTQ.GZ | Raw Reads | Illumina | FALSE | 17927987899 | NA | Breast | NA | NA | Invasive breast carcinoma |\n",
       "| 6 | BCM | ML6061_CAGATCTG_L00X_R2_001.fastq.gz    | 24561 | BCM-0104 | 2011012-104 | NA | BCM-0104 | PDX | RNA-Seq | ⋯ | FASTQ.GZ | Raw Reads | Illumina | FALSE | 18002109303 | NA | Breast | NA | NA | Invasive breast carcinoma |\n",
       "\n"
      ],
      "text/plain": [
       "  ...1 contributor file_name                               patient_id model_id\n",
       "1 1    BCM         LZML_37246_NoIndex_L00X_R1_001.fastq.gz 24452      BCM-0002\n",
       "2 2    BCM         LZML_37246_NoIndex_L00X_R2_001.fastq.gz 24452      BCM-0002\n",
       "3 3    BCM         ML6065_GATCAGCG_L00X_R1_001.fastq.gz    2219       BCM-0046\n",
       "4 4    BCM         ML6065_GATCAGCG_L00X_R2_001.fastq.gz    2219       BCM-0046\n",
       "5 5    BCM         ML6061_CAGATCTG_L00X_R1_001.fastq.gz    24561      BCM-0104\n",
       "6 6    BCM         ML6061_CAGATCTG_L00X_R2_001.fastq.gz    24561      BCM-0104\n",
       "  tumor_id    case_id sample_id sample_type experimental_strategy ⋯ data_format\n",
       "1 2011012-2   NA      BCM-0002  PDX         RNA-Seq               ⋯ FASTQ.GZ   \n",
       "2 2011012-2   NA      BCM-0002  PDX         RNA-Seq               ⋯ FASTQ.GZ   \n",
       "3 2011012-46  NA      BCM-0046  PDX         RNA-Seq               ⋯ FASTQ.GZ   \n",
       "4 2011012-46  NA      BCM-0046  PDX         RNA-Seq               ⋯ FASTQ.GZ   \n",
       "5 2011012-104 NA      BCM-0104  PDX         RNA-Seq               ⋯ FASTQ.GZ   \n",
       "6 2011012-104 NA      BCM-0104  PDX         RNA-Seq               ⋯ FASTQ.GZ   \n",
       "  data_type platform is_ffpe file_size   passage disease_type comment\n",
       "1 Raw Reads Illumina FALSE   20656964441 NA      Breast       NA     \n",
       "2 Raw Reads Illumina FALSE   21093643386 NA      Breast       NA     \n",
       "3 Raw Reads Illumina FALSE   22027304876 NA      Breast       NA     \n",
       "4 Raw Reads Illumina FALSE   22027304876 NA      Breast       NA     \n",
       "5 Raw Reads Illumina FALSE   17927987899 NA      Breast       NA     \n",
       "6 Raw Reads Illumina FALSE   18002109303 NA      Breast       NA     \n",
       "  investigation disease_detail           \n",
       "1 NA            Invasive breast carcinoma\n",
       "2 NA            Invasive breast carcinoma\n",
       "3 NA            Invasive breast carcinoma\n",
       "4 NA            Invasive breast carcinoma\n",
       "5 NA            Invasive breast carcinoma\n",
       "6 NA            Invasive breast carcinoma"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Load the original metadata\n",
    "\n",
    "original_pdtc_metadata <- read_csv('../2022-02-09_pdxnet_pdtc_pdmr_portal_ids/2022-02-09_pdxnet_portal_pdtc_files.csv')\n",
    "\n",
    "nrow(original_pdtc_metadata)\n",
    "head(original_pdtc_metadata)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "2b97ddec-ee01-4692-b032-a6bc3ba2cda6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 29 × 1</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>.</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>...1</th><td>   0</td></tr>\n",
       "\t<tr><th scope=row>contributor</th><td>   0</td></tr>\n",
       "\t<tr><th scope=row>file_name</th><td>   0</td></tr>\n",
       "\t<tr><th scope=row>patient_id</th><td>  26</td></tr>\n",
       "\t<tr><th scope=row>model_id</th><td>1515</td></tr>\n",
       "\t<tr><th scope=row>tumor_id</th><td>2067</td></tr>\n",
       "\t<tr><th scope=row>case_id</th><td> 602</td></tr>\n",
       "\t<tr><th scope=row>sample_id</th><td>   0</td></tr>\n",
       "\t<tr><th scope=row>sample_type</th><td>   0</td></tr>\n",
       "\t<tr><th scope=row>experimental_strategy</th><td>   0</td></tr>\n",
       "\t<tr><th scope=row>gender</th><td> 436</td></tr>\n",
       "\t<tr><th scope=row>paired_end</th><td>  17</td></tr>\n",
       "\t<tr><th scope=row>capture_kit</th><td> 821</td></tr>\n",
       "\t<tr><th scope=row>capture_assembly</th><td>2205</td></tr>\n",
       "\t<tr><th scope=row>availability</th><td>   0</td></tr>\n",
       "\t<tr><th scope=row>public</th><td>  48</td></tr>\n",
       "\t<tr><th scope=row>access_level</th><td>1345</td></tr>\n",
       "\t<tr><th scope=row>created_datetime</th><td>2711</td></tr>\n",
       "\t<tr><th scope=row>data_category</th><td>1345</td></tr>\n",
       "\t<tr><th scope=row>data_format</th><td>1297</td></tr>\n",
       "\t<tr><th scope=row>data_type</th><td>1297</td></tr>\n",
       "\t<tr><th scope=row>platform</th><td> 124</td></tr>\n",
       "\t<tr><th scope=row>is_ffpe</th><td>2356</td></tr>\n",
       "\t<tr><th scope=row>file_size</th><td>1403</td></tr>\n",
       "\t<tr><th scope=row>passage</th><td>1631</td></tr>\n",
       "\t<tr><th scope=row>disease_type</th><td> 515</td></tr>\n",
       "\t<tr><th scope=row>comment</th><td>2776</td></tr>\n",
       "\t<tr><th scope=row>investigation</th><td>2822</td></tr>\n",
       "\t<tr><th scope=row>disease_detail</th><td> 515</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 29 × 1\n",
       "\\begin{tabular}{r|l}\n",
       "  & .\\\\\n",
       "  & <dbl>\\\\\n",
       "\\hline\n",
       "\t...1 &    0\\\\\n",
       "\tcontributor &    0\\\\\n",
       "\tfile\\_name &    0\\\\\n",
       "\tpatient\\_id &   26\\\\\n",
       "\tmodel\\_id & 1515\\\\\n",
       "\ttumor\\_id & 2067\\\\\n",
       "\tcase\\_id &  602\\\\\n",
       "\tsample\\_id &    0\\\\\n",
       "\tsample\\_type &    0\\\\\n",
       "\texperimental\\_strategy &    0\\\\\n",
       "\tgender &  436\\\\\n",
       "\tpaired\\_end &   17\\\\\n",
       "\tcapture\\_kit &  821\\\\\n",
       "\tcapture\\_assembly & 2205\\\\\n",
       "\tavailability &    0\\\\\n",
       "\tpublic &   48\\\\\n",
       "\taccess\\_level & 1345\\\\\n",
       "\tcreated\\_datetime & 2711\\\\\n",
       "\tdata\\_category & 1345\\\\\n",
       "\tdata\\_format & 1297\\\\\n",
       "\tdata\\_type & 1297\\\\\n",
       "\tplatform &  124\\\\\n",
       "\tis\\_ffpe & 2356\\\\\n",
       "\tfile\\_size & 1403\\\\\n",
       "\tpassage & 1631\\\\\n",
       "\tdisease\\_type &  515\\\\\n",
       "\tcomment & 2776\\\\\n",
       "\tinvestigation & 2822\\\\\n",
       "\tdisease\\_detail &  515\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 29 × 1\n",
       "\n",
       "| <!--/--> | . &lt;dbl&gt; |\n",
       "|---|---|\n",
       "| ...1 |    0 |\n",
       "| contributor |    0 |\n",
       "| file_name |    0 |\n",
       "| patient_id |   26 |\n",
       "| model_id | 1515 |\n",
       "| tumor_id | 2067 |\n",
       "| case_id |  602 |\n",
       "| sample_id |    0 |\n",
       "| sample_type |    0 |\n",
       "| experimental_strategy |    0 |\n",
       "| gender |  436 |\n",
       "| paired_end |   17 |\n",
       "| capture_kit |  821 |\n",
       "| capture_assembly | 2205 |\n",
       "| availability |    0 |\n",
       "| public |   48 |\n",
       "| access_level | 1345 |\n",
       "| created_datetime | 2711 |\n",
       "| data_category | 1345 |\n",
       "| data_format | 1297 |\n",
       "| data_type | 1297 |\n",
       "| platform |  124 |\n",
       "| is_ffpe | 2356 |\n",
       "| file_size | 1403 |\n",
       "| passage | 1631 |\n",
       "| disease_type |  515 |\n",
       "| comment | 2776 |\n",
       "| investigation | 2822 |\n",
       "| disease_detail |  515 |\n",
       "\n"
      ],
      "text/plain": [
       "                      .   \n",
       "...1                     0\n",
       "contributor              0\n",
       "file_name                0\n",
       "patient_id              26\n",
       "model_id              1515\n",
       "tumor_id              2067\n",
       "case_id                602\n",
       "sample_id                0\n",
       "sample_type              0\n",
       "experimental_strategy    0\n",
       "gender                 436\n",
       "paired_end              17\n",
       "capture_kit            821\n",
       "capture_assembly      2205\n",
       "availability             0\n",
       "public                  48\n",
       "access_level          1345\n",
       "created_datetime      2711\n",
       "data_category         1345\n",
       "data_format           1297\n",
       "data_type             1297\n",
       "platform               124\n",
       "is_ffpe               2356\n",
       "file_size             1403\n",
       "passage               1631\n",
       "disease_type           515\n",
       "comment               2776\n",
       "investigation         2822\n",
       "disease_detail         515"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# NAs and missing data originally?\n",
    "\n",
    "summarize_missing_fields(original_pdtc_metadata)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "b9a6ca5f-724c-40b7-a100-2c0615fccdc4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<dl>\n",
       "\t<dt>$BCM</dt>\n",
       "\t\t<dd><table class=\"dataframe\">\n",
       "<caption>A tibble: 1 × 29</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>...1</th><th scope=col>contributor</th><th scope=col>file_name</th><th scope=col>patient_id</th><th scope=col>model_id</th><th scope=col>tumor_id</th><th scope=col>case_id</th><th scope=col>sample_id</th><th scope=col>sample_type</th><th scope=col>experimental_strategy</th><th scope=col>⋯</th><th scope=col>data_format</th><th scope=col>data_type</th><th scope=col>platform</th><th scope=col>is_ffpe</th><th scope=col>file_size</th><th scope=col>passage</th><th scope=col>disease_type</th><th scope=col>comment</th><th scope=col>investigation</th><th scope=col>disease_detail</th></tr>\n",
       "\t<tr><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td>BCM</td><td>LZML_37246_NoIndex_L00X_R1_001.fastq.gz</td><td>24452</td><td>BCM-0002</td><td>2011012-2</td><td>NA</td><td>BCM-0002</td><td>PDX</td><td>RNA-Seq</td><td>⋯</td><td>FASTQ.GZ</td><td>Raw Reads</td><td>Illumina</td><td>FALSE</td><td>20656964441</td><td>NA</td><td>Breast</td><td>NA</td><td>NA</td><td>Invasive breast carcinoma</td></tr>\n",
       "</tbody>\n",
       "</table>\n",
       "</dd>\n",
       "\t<dt>$HCI</dt>\n",
       "\t\t<dd><table class=\"dataframe\">\n",
       "<caption>A tibble: 1 × 29</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>...1</th><th scope=col>contributor</th><th scope=col>file_name</th><th scope=col>patient_id</th><th scope=col>model_id</th><th scope=col>tumor_id</th><th scope=col>case_id</th><th scope=col>sample_id</th><th scope=col>sample_type</th><th scope=col>experimental_strategy</th><th scope=col>⋯</th><th scope=col>data_format</th><th scope=col>data_type</th><th scope=col>platform</th><th scope=col>is_ffpe</th><th scope=col>file_size</th><th scope=col>passage</th><th scope=col>disease_type</th><th scope=col>comment</th><th scope=col>investigation</th><th scope=col>disease_detail</th></tr>\n",
       "\t<tr><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>209</td><td>HCI</td><td>14990X12_HCI001_patientnormal_1.txt.gz</td><td>HCI001</td><td>NA</td><td>NA</td><td>HCI001</td><td>14990X12</td><td>Normal</td><td>WES</td><td>⋯</td><td>NA</td><td>NA</td><td>Illumina HiSeq</td><td>NA</td><td>NA</td><td>NA</td><td>Breast</td><td>NA</td><td>NA</td><td>Breast Poorly Differentiated</td></tr>\n",
       "</tbody>\n",
       "</table>\n",
       "</dd>\n",
       "\t<dt>$MDACC</dt>\n",
       "\t\t<dd><table class=\"dataframe\">\n",
       "<caption>A tibble: 1 × 29</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>...1</th><th scope=col>contributor</th><th scope=col>file_name</th><th scope=col>patient_id</th><th scope=col>model_id</th><th scope=col>tumor_id</th><th scope=col>case_id</th><th scope=col>sample_id</th><th scope=col>sample_type</th><th scope=col>experimental_strategy</th><th scope=col>⋯</th><th scope=col>data_format</th><th scope=col>data_type</th><th scope=col>platform</th><th scope=col>is_ffpe</th><th scope=col>file_size</th><th scope=col>passage</th><th scope=col>disease_type</th><th scope=col>comment</th><th scope=col>investigation</th><th scope=col>disease_detail</th></tr>\n",
       "\t<tr><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1100</td><td>MDACC</td><td>BFang-LungPDX-MS11DNA-TC211W.R1.fastq.gz</td><td>TC211</td><td>NA</td><td>NA</td><td>TC211</td><td>TC211W</td><td>Normal</td><td>WES</td><td>⋯</td><td>NA</td><td>NA</td><td>Illumina Hiseq</td><td>NA</td><td>NA</td><td>NA</td><td>Blood</td><td>NA</td><td>NA</td><td>Blood</td></tr>\n",
       "</tbody>\n",
       "</table>\n",
       "</dd>\n",
       "\t<dt>$`UC Davis`</dt>\n",
       "\t\t<dd><table class=\"dataframe\">\n",
       "<caption>A tibble: 1 × 29</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>...1</th><th scope=col>contributor</th><th scope=col>file_name</th><th scope=col>patient_id</th><th scope=col>model_id</th><th scope=col>tumor_id</th><th scope=col>case_id</th><th scope=col>sample_id</th><th scope=col>sample_type</th><th scope=col>experimental_strategy</th><th scope=col>⋯</th><th scope=col>data_format</th><th scope=col>data_type</th><th scope=col>platform</th><th scope=col>is_ffpe</th><th scope=col>file_size</th><th scope=col>passage</th><th scope=col>disease_type</th><th scope=col>comment</th><th scope=col>investigation</th><th scope=col>disease_detail</th></tr>\n",
       "\t<tr><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1518</td><td>UC Davis</td><td>BL0269F404_P0_GES10_166_ATCACG_L004_R1_ALL.fastq</td><td>BL0269F</td><td>BL0269F</td><td>BL0269F</td><td>BL0269F</td><td>BL0269F</td><td>Tumor</td><td>RNA-Seq</td><td>⋯</td><td>fastq</td><td>raw data</td><td>Illumina</td><td>NA</td><td>NA</td><td>P0</td><td>Bladder</td><td>NA</td><td>NA</td><td>bladder urothelial carcinoma</td></tr>\n",
       "</tbody>\n",
       "</table>\n",
       "</dd>\n",
       "\t<dt>$WISTAR</dt>\n",
       "\t\t<dd><table class=\"dataframe\">\n",
       "<caption>A tibble: 1 × 29</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>...1</th><th scope=col>contributor</th><th scope=col>file_name</th><th scope=col>patient_id</th><th scope=col>model_id</th><th scope=col>tumor_id</th><th scope=col>case_id</th><th scope=col>sample_id</th><th scope=col>sample_type</th><th scope=col>experimental_strategy</th><th scope=col>⋯</th><th scope=col>data_format</th><th scope=col>data_type</th><th scope=col>platform</th><th scope=col>is_ffpe</th><th scope=col>file_size</th><th scope=col>passage</th><th scope=col>disease_type</th><th scope=col>comment</th><th scope=col>investigation</th><th scope=col>disease_detail</th></tr>\n",
       "\t<tr><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1566</td><td>WISTAR</td><td>wm3907n_R1.fastq.gz</td><td>WM3907</td><td>wm3907n</td><td>NA</td><td>NA</td><td>wm3907n</td><td>Tumor</td><td>WES</td><td>⋯</td><td>fastq.gz</td><td>Raw Reads</td><td>Illumina</td><td>FALSE</td><td>5439234</td><td>Tumor</td><td>Unknown</td><td>NA</td><td>NA</td><td>Unknown</td></tr>\n",
       "</tbody>\n",
       "</table>\n",
       "</dd>\n",
       "\t<dt>$WUSTL</dt>\n",
       "\t\t<dd><table class=\"dataframe\">\n",
       "<caption>A tibble: 1 × 29</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>...1</th><th scope=col>contributor</th><th scope=col>file_name</th><th scope=col>patient_id</th><th scope=col>model_id</th><th scope=col>tumor_id</th><th scope=col>case_id</th><th scope=col>sample_id</th><th scope=col>sample_type</th><th scope=col>experimental_strategy</th><th scope=col>⋯</th><th scope=col>data_format</th><th scope=col>data_type</th><th scope=col>platform</th><th scope=col>is_ffpe</th><th scope=col>file_size</th><th scope=col>passage</th><th scope=col>disease_type</th><th scope=col>comment</th><th scope=col>investigation</th><th scope=col>disease_detail</th></tr>\n",
       "\t<tr><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1948</td><td>WUSTL</td><td>172845-121-T_SL.exome.cram</td><td>PDMR-172845</td><td>172845-121-T_SL</td><td>121-T</td><td>PDMR-172845</td><td>172845-121-T_SL</td><td>PDX</td><td>WES</td><td>⋯</td><td>cram</td><td>Aligned Reads</td><td>NovaSeq S4</td><td>NA</td><td>1681781885</td><td>P3</td><td>Unknown</td><td>NA</td><td>NA</td><td>Unknown</td></tr>\n",
       "</tbody>\n",
       "</table>\n",
       "</dd>\n",
       "</dl>\n"
      ],
      "text/latex": [
       "\\begin{description}\n",
       "\\item[\\$BCM] A tibble: 1 × 29\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " ...1 & contributor & file\\_name & patient\\_id & model\\_id & tumor\\_id & case\\_id & sample\\_id & sample\\_type & experimental\\_strategy & ⋯ & data\\_format & data\\_type & platform & is\\_ffpe & file\\_size & passage & disease\\_type & comment & investigation & disease\\_detail\\\\\n",
       " <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & ⋯ & <chr> & <chr> & <chr> & <lgl> & <dbl> & <chr> & <chr> & <chr> & <lgl> & <chr>\\\\\n",
       "\\hline\n",
       "\t 1 & BCM & LZML\\_37246\\_NoIndex\\_L00X\\_R1\\_001.fastq.gz & 24452 & BCM-0002 & 2011012-2 & NA & BCM-0002 & PDX & RNA-Seq & ⋯ & FASTQ.GZ & Raw Reads & Illumina & FALSE & 20656964441 & NA & Breast & NA & NA & Invasive breast carcinoma\\\\\n",
       "\\end{tabular}\n",
       "\n",
       "\\item[\\$HCI] A tibble: 1 × 29\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " ...1 & contributor & file\\_name & patient\\_id & model\\_id & tumor\\_id & case\\_id & sample\\_id & sample\\_type & experimental\\_strategy & ⋯ & data\\_format & data\\_type & platform & is\\_ffpe & file\\_size & passage & disease\\_type & comment & investigation & disease\\_detail\\\\\n",
       " <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & ⋯ & <chr> & <chr> & <chr> & <lgl> & <dbl> & <chr> & <chr> & <chr> & <lgl> & <chr>\\\\\n",
       "\\hline\n",
       "\t 209 & HCI & 14990X12\\_HCI001\\_patientnormal\\_1.txt.gz & HCI001 & NA & NA & HCI001 & 14990X12 & Normal & WES & ⋯ & NA & NA & Illumina HiSeq & NA & NA & NA & Breast & NA & NA & Breast Poorly Differentiated\\\\\n",
       "\\end{tabular}\n",
       "\n",
       "\\item[\\$MDACC] A tibble: 1 × 29\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " ...1 & contributor & file\\_name & patient\\_id & model\\_id & tumor\\_id & case\\_id & sample\\_id & sample\\_type & experimental\\_strategy & ⋯ & data\\_format & data\\_type & platform & is\\_ffpe & file\\_size & passage & disease\\_type & comment & investigation & disease\\_detail\\\\\n",
       " <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & ⋯ & <chr> & <chr> & <chr> & <lgl> & <dbl> & <chr> & <chr> & <chr> & <lgl> & <chr>\\\\\n",
       "\\hline\n",
       "\t 1100 & MDACC & BFang-LungPDX-MS11DNA-TC211W.R1.fastq.gz & TC211 & NA & NA & TC211 & TC211W & Normal & WES & ⋯ & NA & NA & Illumina Hiseq & NA & NA & NA & Blood & NA & NA & Blood\\\\\n",
       "\\end{tabular}\n",
       "\n",
       "\\item[\\$`UC Davis`] A tibble: 1 × 29\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " ...1 & contributor & file\\_name & patient\\_id & model\\_id & tumor\\_id & case\\_id & sample\\_id & sample\\_type & experimental\\_strategy & ⋯ & data\\_format & data\\_type & platform & is\\_ffpe & file\\_size & passage & disease\\_type & comment & investigation & disease\\_detail\\\\\n",
       " <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & ⋯ & <chr> & <chr> & <chr> & <lgl> & <dbl> & <chr> & <chr> & <chr> & <lgl> & <chr>\\\\\n",
       "\\hline\n",
       "\t 1518 & UC Davis & BL0269F404\\_P0\\_GES10\\_166\\_ATCACG\\_L004\\_R1\\_ALL.fastq & BL0269F & BL0269F & BL0269F & BL0269F & BL0269F & Tumor & RNA-Seq & ⋯ & fastq & raw data & Illumina & NA & NA & P0 & Bladder & NA & NA & bladder urothelial carcinoma\\\\\n",
       "\\end{tabular}\n",
       "\n",
       "\\item[\\$WISTAR] A tibble: 1 × 29\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " ...1 & contributor & file\\_name & patient\\_id & model\\_id & tumor\\_id & case\\_id & sample\\_id & sample\\_type & experimental\\_strategy & ⋯ & data\\_format & data\\_type & platform & is\\_ffpe & file\\_size & passage & disease\\_type & comment & investigation & disease\\_detail\\\\\n",
       " <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & ⋯ & <chr> & <chr> & <chr> & <lgl> & <dbl> & <chr> & <chr> & <chr> & <lgl> & <chr>\\\\\n",
       "\\hline\n",
       "\t 1566 & WISTAR & wm3907n\\_R1.fastq.gz & WM3907 & wm3907n & NA & NA & wm3907n & Tumor & WES & ⋯ & fastq.gz & Raw Reads & Illumina & FALSE & 5439234 & Tumor & Unknown & NA & NA & Unknown\\\\\n",
       "\\end{tabular}\n",
       "\n",
       "\\item[\\$WUSTL] A tibble: 1 × 29\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " ...1 & contributor & file\\_name & patient\\_id & model\\_id & tumor\\_id & case\\_id & sample\\_id & sample\\_type & experimental\\_strategy & ⋯ & data\\_format & data\\_type & platform & is\\_ffpe & file\\_size & passage & disease\\_type & comment & investigation & disease\\_detail\\\\\n",
       " <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & ⋯ & <chr> & <chr> & <chr> & <lgl> & <dbl> & <chr> & <chr> & <chr> & <lgl> & <chr>\\\\\n",
       "\\hline\n",
       "\t 1948 & WUSTL & 172845-121-T\\_SL.exome.cram & PDMR-172845 & 172845-121-T\\_SL & 121-T & PDMR-172845 & 172845-121-T\\_SL & PDX & WES & ⋯ & cram & Aligned Reads & NovaSeq S4 & NA & 1681781885 & P3 & Unknown & NA & NA & Unknown\\\\\n",
       "\\end{tabular}\n",
       "\n",
       "\\end{description}\n"
      ],
      "text/markdown": [
       "$BCM\n",
       ":   \n",
       "A tibble: 1 × 29\n",
       "\n",
       "| ...1 &lt;dbl&gt; | contributor &lt;chr&gt; | file_name &lt;chr&gt; | patient_id &lt;chr&gt; | model_id &lt;chr&gt; | tumor_id &lt;chr&gt; | case_id &lt;chr&gt; | sample_id &lt;chr&gt; | sample_type &lt;chr&gt; | experimental_strategy &lt;chr&gt; | ⋯ ⋯ | data_format &lt;chr&gt; | data_type &lt;chr&gt; | platform &lt;chr&gt; | is_ffpe &lt;lgl&gt; | file_size &lt;dbl&gt; | passage &lt;chr&gt; | disease_type &lt;chr&gt; | comment &lt;chr&gt; | investigation &lt;lgl&gt; | disease_detail &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | BCM | LZML_37246_NoIndex_L00X_R1_001.fastq.gz | 24452 | BCM-0002 | 2011012-2 | NA | BCM-0002 | PDX | RNA-Seq | ⋯ | FASTQ.GZ | Raw Reads | Illumina | FALSE | 20656964441 | NA | Breast | NA | NA | Invasive breast carcinoma |\n",
       "\n",
       "\n",
       "$HCI\n",
       ":   \n",
       "A tibble: 1 × 29\n",
       "\n",
       "| ...1 &lt;dbl&gt; | contributor &lt;chr&gt; | file_name &lt;chr&gt; | patient_id &lt;chr&gt; | model_id &lt;chr&gt; | tumor_id &lt;chr&gt; | case_id &lt;chr&gt; | sample_id &lt;chr&gt; | sample_type &lt;chr&gt; | experimental_strategy &lt;chr&gt; | ⋯ ⋯ | data_format &lt;chr&gt; | data_type &lt;chr&gt; | platform &lt;chr&gt; | is_ffpe &lt;lgl&gt; | file_size &lt;dbl&gt; | passage &lt;chr&gt; | disease_type &lt;chr&gt; | comment &lt;chr&gt; | investigation &lt;lgl&gt; | disease_detail &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 209 | HCI | 14990X12_HCI001_patientnormal_1.txt.gz | HCI001 | NA | NA | HCI001 | 14990X12 | Normal | WES | ⋯ | NA | NA | Illumina HiSeq | NA | NA | NA | Breast | NA | NA | Breast Poorly Differentiated |\n",
       "\n",
       "\n",
       "$MDACC\n",
       ":   \n",
       "A tibble: 1 × 29\n",
       "\n",
       "| ...1 &lt;dbl&gt; | contributor &lt;chr&gt; | file_name &lt;chr&gt; | patient_id &lt;chr&gt; | model_id &lt;chr&gt; | tumor_id &lt;chr&gt; | case_id &lt;chr&gt; | sample_id &lt;chr&gt; | sample_type &lt;chr&gt; | experimental_strategy &lt;chr&gt; | ⋯ ⋯ | data_format &lt;chr&gt; | data_type &lt;chr&gt; | platform &lt;chr&gt; | is_ffpe &lt;lgl&gt; | file_size &lt;dbl&gt; | passage &lt;chr&gt; | disease_type &lt;chr&gt; | comment &lt;chr&gt; | investigation &lt;lgl&gt; | disease_detail &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1100 | MDACC | BFang-LungPDX-MS11DNA-TC211W.R1.fastq.gz | TC211 | NA | NA | TC211 | TC211W | Normal | WES | ⋯ | NA | NA | Illumina Hiseq | NA | NA | NA | Blood | NA | NA | Blood |\n",
       "\n",
       "\n",
       "$`UC Davis`\n",
       ":   \n",
       "A tibble: 1 × 29\n",
       "\n",
       "| ...1 &lt;dbl&gt; | contributor &lt;chr&gt; | file_name &lt;chr&gt; | patient_id &lt;chr&gt; | model_id &lt;chr&gt; | tumor_id &lt;chr&gt; | case_id &lt;chr&gt; | sample_id &lt;chr&gt; | sample_type &lt;chr&gt; | experimental_strategy &lt;chr&gt; | ⋯ ⋯ | data_format &lt;chr&gt; | data_type &lt;chr&gt; | platform &lt;chr&gt; | is_ffpe &lt;lgl&gt; | file_size &lt;dbl&gt; | passage &lt;chr&gt; | disease_type &lt;chr&gt; | comment &lt;chr&gt; | investigation &lt;lgl&gt; | disease_detail &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1518 | UC Davis | BL0269F404_P0_GES10_166_ATCACG_L004_R1_ALL.fastq | BL0269F | BL0269F | BL0269F | BL0269F | BL0269F | Tumor | RNA-Seq | ⋯ | fastq | raw data | Illumina | NA | NA | P0 | Bladder | NA | NA | bladder urothelial carcinoma |\n",
       "\n",
       "\n",
       "$WISTAR\n",
       ":   \n",
       "A tibble: 1 × 29\n",
       "\n",
       "| ...1 &lt;dbl&gt; | contributor &lt;chr&gt; | file_name &lt;chr&gt; | patient_id &lt;chr&gt; | model_id &lt;chr&gt; | tumor_id &lt;chr&gt; | case_id &lt;chr&gt; | sample_id &lt;chr&gt; | sample_type &lt;chr&gt; | experimental_strategy &lt;chr&gt; | ⋯ ⋯ | data_format &lt;chr&gt; | data_type &lt;chr&gt; | platform &lt;chr&gt; | is_ffpe &lt;lgl&gt; | file_size &lt;dbl&gt; | passage &lt;chr&gt; | disease_type &lt;chr&gt; | comment &lt;chr&gt; | investigation &lt;lgl&gt; | disease_detail &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1566 | WISTAR | wm3907n_R1.fastq.gz | WM3907 | wm3907n | NA | NA | wm3907n | Tumor | WES | ⋯ | fastq.gz | Raw Reads | Illumina | FALSE | 5439234 | Tumor | Unknown | NA | NA | Unknown |\n",
       "\n",
       "\n",
       "$WUSTL\n",
       ":   \n",
       "A tibble: 1 × 29\n",
       "\n",
       "| ...1 &lt;dbl&gt; | contributor &lt;chr&gt; | file_name &lt;chr&gt; | patient_id &lt;chr&gt; | model_id &lt;chr&gt; | tumor_id &lt;chr&gt; | case_id &lt;chr&gt; | sample_id &lt;chr&gt; | sample_type &lt;chr&gt; | experimental_strategy &lt;chr&gt; | ⋯ ⋯ | data_format &lt;chr&gt; | data_type &lt;chr&gt; | platform &lt;chr&gt; | is_ffpe &lt;lgl&gt; | file_size &lt;dbl&gt; | passage &lt;chr&gt; | disease_type &lt;chr&gt; | comment &lt;chr&gt; | investigation &lt;lgl&gt; | disease_detail &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1948 | WUSTL | 172845-121-T_SL.exome.cram | PDMR-172845 | 172845-121-T_SL | 121-T | PDMR-172845 | 172845-121-T_SL | PDX | WES | ⋯ | cram | Aligned Reads | NovaSeq S4 | NA | 1681781885 | P3 | Unknown | NA | NA | Unknown |\n",
       "\n",
       "\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "$BCM\n",
       "\u001b[90m# A tibble: 1 × 29\u001b[39m\n",
       "   ...1 contributor file_name     patient_id model_id tumor_id case_id sample_id\n",
       "  \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m         \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m      \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m    \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m    \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m   \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m    \n",
       "\u001b[90m1\u001b[39m     1 BCM         LZML_37246_N… 24452      BCM-0002 2011012… \u001b[31mNA\u001b[39m      BCM-0002 \n",
       "\u001b[90m# … with 21 more variables: sample_type <chr>, experimental_strategy <chr>,\u001b[39m\n",
       "\u001b[90m#   gender <chr>, paired_end <dbl>, capture_kit <chr>, capture_assembly <chr>,\u001b[39m\n",
       "\u001b[90m#   availability <chr>, public <chr>, access_level <chr>,\u001b[39m\n",
       "\u001b[90m#   created_datetime <dbl>, data_category <chr>, data_format <chr>,\u001b[39m\n",
       "\u001b[90m#   data_type <chr>, platform <chr>, is_ffpe <lgl>, file_size <dbl>,\u001b[39m\n",
       "\u001b[90m#   passage <chr>, disease_type <chr>, comment <chr>, investigation <lgl>,\u001b[39m\n",
       "\u001b[90m#   disease_detail <chr>\u001b[39m\n",
       "\n",
       "$HCI\n",
       "\u001b[90m# A tibble: 1 × 29\u001b[39m\n",
       "   ...1 contributor file_name     patient_id model_id tumor_id case_id sample_id\n",
       "  \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m         \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m      \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m    \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m    \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m   \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m    \n",
       "\u001b[90m1\u001b[39m   209 HCI         14990X12_HCI… HCI001     \u001b[31mNA\u001b[39m       \u001b[31mNA\u001b[39m       HCI001  14990X12 \n",
       "\u001b[90m# … with 21 more variables: sample_type <chr>, experimental_strategy <chr>,\u001b[39m\n",
       "\u001b[90m#   gender <chr>, paired_end <dbl>, capture_kit <chr>, capture_assembly <chr>,\u001b[39m\n",
       "\u001b[90m#   availability <chr>, public <chr>, access_level <chr>,\u001b[39m\n",
       "\u001b[90m#   created_datetime <dbl>, data_category <chr>, data_format <chr>,\u001b[39m\n",
       "\u001b[90m#   data_type <chr>, platform <chr>, is_ffpe <lgl>, file_size <dbl>,\u001b[39m\n",
       "\u001b[90m#   passage <chr>, disease_type <chr>, comment <chr>, investigation <lgl>,\u001b[39m\n",
       "\u001b[90m#   disease_detail <chr>\u001b[39m\n",
       "\n",
       "$MDACC\n",
       "\u001b[90m# A tibble: 1 × 29\u001b[39m\n",
       "   ...1 contributor file_name     patient_id model_id tumor_id case_id sample_id\n",
       "  \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m         \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m      \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m    \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m    \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m   \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m    \n",
       "\u001b[90m1\u001b[39m  \u001b[4m1\u001b[24m100 MDACC       BFang-LungPD… TC211      \u001b[31mNA\u001b[39m       \u001b[31mNA\u001b[39m       TC211   TC211W   \n",
       "\u001b[90m# … with 21 more variables: sample_type <chr>, experimental_strategy <chr>,\u001b[39m\n",
       "\u001b[90m#   gender <chr>, paired_end <dbl>, capture_kit <chr>, capture_assembly <chr>,\u001b[39m\n",
       "\u001b[90m#   availability <chr>, public <chr>, access_level <chr>,\u001b[39m\n",
       "\u001b[90m#   created_datetime <dbl>, data_category <chr>, data_format <chr>,\u001b[39m\n",
       "\u001b[90m#   data_type <chr>, platform <chr>, is_ffpe <lgl>, file_size <dbl>,\u001b[39m\n",
       "\u001b[90m#   passage <chr>, disease_type <chr>, comment <chr>, investigation <lgl>,\u001b[39m\n",
       "\u001b[90m#   disease_detail <chr>\u001b[39m\n",
       "\n",
       "$`UC Davis`\n",
       "\u001b[90m# A tibble: 1 × 29\u001b[39m\n",
       "   ...1 contributor file_name     patient_id model_id tumor_id case_id sample_id\n",
       "  \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m         \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m      \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m    \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m    \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m   \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m    \n",
       "\u001b[90m1\u001b[39m  \u001b[4m1\u001b[24m518 UC Davis    BL0269F404_P… BL0269F    BL0269F  BL0269F  BL0269F BL0269F  \n",
       "\u001b[90m# … with 21 more variables: sample_type <chr>, experimental_strategy <chr>,\u001b[39m\n",
       "\u001b[90m#   gender <chr>, paired_end <dbl>, capture_kit <chr>, capture_assembly <chr>,\u001b[39m\n",
       "\u001b[90m#   availability <chr>, public <chr>, access_level <chr>,\u001b[39m\n",
       "\u001b[90m#   created_datetime <dbl>, data_category <chr>, data_format <chr>,\u001b[39m\n",
       "\u001b[90m#   data_type <chr>, platform <chr>, is_ffpe <lgl>, file_size <dbl>,\u001b[39m\n",
       "\u001b[90m#   passage <chr>, disease_type <chr>, comment <chr>, investigation <lgl>,\u001b[39m\n",
       "\u001b[90m#   disease_detail <chr>\u001b[39m\n",
       "\n",
       "$WISTAR\n",
       "\u001b[90m# A tibble: 1 × 29\u001b[39m\n",
       "   ...1 contributor file_name     patient_id model_id tumor_id case_id sample_id\n",
       "  \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m         \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m      \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m    \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m    \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m   \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m    \n",
       "\u001b[90m1\u001b[39m  \u001b[4m1\u001b[24m566 WISTAR      wm3907n_R1.f… WM3907     wm3907n  \u001b[31mNA\u001b[39m       \u001b[31mNA\u001b[39m      wm3907n  \n",
       "\u001b[90m# … with 21 more variables: sample_type <chr>, experimental_strategy <chr>,\u001b[39m\n",
       "\u001b[90m#   gender <chr>, paired_end <dbl>, capture_kit <chr>, capture_assembly <chr>,\u001b[39m\n",
       "\u001b[90m#   availability <chr>, public <chr>, access_level <chr>,\u001b[39m\n",
       "\u001b[90m#   created_datetime <dbl>, data_category <chr>, data_format <chr>,\u001b[39m\n",
       "\u001b[90m#   data_type <chr>, platform <chr>, is_ffpe <lgl>, file_size <dbl>,\u001b[39m\n",
       "\u001b[90m#   passage <chr>, disease_type <chr>, comment <chr>, investigation <lgl>,\u001b[39m\n",
       "\u001b[90m#   disease_detail <chr>\u001b[39m\n",
       "\n",
       "$WUSTL\n",
       "\u001b[90m# A tibble: 1 × 29\u001b[39m\n",
       "   ...1 contributor file_name   patient_id  model_id  tumor_id case_id sample_id\n",
       "  \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m     \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m    \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m   \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m    \n",
       "\u001b[90m1\u001b[39m  \u001b[4m1\u001b[24m948 WUSTL       172845-121… PDMR-172845 172845-1… 121-T    PDMR-1… 172845-1…\n",
       "\u001b[90m# … with 21 more variables: sample_type <chr>, experimental_strategy <chr>,\u001b[39m\n",
       "\u001b[90m#   gender <chr>, paired_end <dbl>, capture_kit <chr>, capture_assembly <chr>,\u001b[39m\n",
       "\u001b[90m#   availability <chr>, public <chr>, access_level <chr>,\u001b[39m\n",
       "\u001b[90m#   created_datetime <dbl>, data_category <chr>, data_format <chr>,\u001b[39m\n",
       "\u001b[90m#   data_type <chr>, platform <chr>, is_ffpe <lgl>, file_size <dbl>,\u001b[39m\n",
       "\u001b[90m#   passage <chr>, disease_type <chr>, comment <chr>, investigation <lgl>,\u001b[39m\n",
       "\u001b[90m#   disease_detail <chr>\u001b[39m\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Break down by PDTC\n",
    "\n",
    "original_pdtc_metadata <- original_pdtc_metadata %>% group_split(contributor)\n",
    "names(original_pdtc_metadata) <- c('BCM', 'HCI', 'MDACC', 'UC Davis', 'WISTAR', 'WUSTL')\n",
    "\n",
    "original_pdtc_metadata %>% map(head, 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "1b490593-7672-4281-8109-80a1025c71b6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<dl>\n",
       "\t<dt>$BCM</dt>\n",
       "\t\t<dd><table class=\"dataframe\">\n",
       "<caption>A data.frame: 29 × 1</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>.</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>...1</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>contributor</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>file_name</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>patient_id</th><td>  2</td></tr>\n",
       "\t<tr><th scope=row>model_id</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>tumor_id</th><td> 16</td></tr>\n",
       "\t<tr><th scope=row>case_id</th><td>208</td></tr>\n",
       "\t<tr><th scope=row>sample_id</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>sample_type</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>experimental_strategy</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>gender</th><td> 42</td></tr>\n",
       "\t<tr><th scope=row>paired_end</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>capture_kit</th><td>124</td></tr>\n",
       "\t<tr><th scope=row>capture_assembly</th><td>208</td></tr>\n",
       "\t<tr><th scope=row>availability</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>public</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>access_level</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>created_datetime</th><td>124</td></tr>\n",
       "\t<tr><th scope=row>data_category</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>data_format</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>data_type</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>platform</th><td>124</td></tr>\n",
       "\t<tr><th scope=row>is_ffpe</th><td>124</td></tr>\n",
       "\t<tr><th scope=row>file_size</th><td> 46</td></tr>\n",
       "\t<tr><th scope=row>passage</th><td>208</td></tr>\n",
       "\t<tr><th scope=row>disease_type</th><td> 98</td></tr>\n",
       "\t<tr><th scope=row>comment</th><td>208</td></tr>\n",
       "\t<tr><th scope=row>investigation</th><td>208</td></tr>\n",
       "\t<tr><th scope=row>disease_detail</th><td> 98</td></tr>\n",
       "</tbody>\n",
       "</table>\n",
       "</dd>\n",
       "\t<dt>$HCI</dt>\n",
       "\t\t<dd><table class=\"dataframe\">\n",
       "<caption>A data.frame: 29 × 1</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>.</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>...1</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>contributor</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>file_name</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>patient_id</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>model_id</th><td>891</td></tr>\n",
       "\t<tr><th scope=row>tumor_id</th><td>891</td></tr>\n",
       "\t<tr><th scope=row>case_id</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>sample_id</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>sample_type</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>experimental_strategy</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>gender</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>paired_end</th><td>  8</td></tr>\n",
       "\t<tr><th scope=row>capture_kit</th><td>697</td></tr>\n",
       "\t<tr><th scope=row>capture_assembly</th><td>683</td></tr>\n",
       "\t<tr><th scope=row>availability</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>public</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>access_level</th><td>891</td></tr>\n",
       "\t<tr><th scope=row>created_datetime</th><td>891</td></tr>\n",
       "\t<tr><th scope=row>data_category</th><td>891</td></tr>\n",
       "\t<tr><th scope=row>data_format</th><td>891</td></tr>\n",
       "\t<tr><th scope=row>data_type</th><td>891</td></tr>\n",
       "\t<tr><th scope=row>platform</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>is_ffpe</th><td>891</td></tr>\n",
       "\t<tr><th scope=row>file_size</th><td>891</td></tr>\n",
       "\t<tr><th scope=row>passage</th><td>779</td></tr>\n",
       "\t<tr><th scope=row>disease_type</th><td>  8</td></tr>\n",
       "\t<tr><th scope=row>comment</th><td>859</td></tr>\n",
       "\t<tr><th scope=row>investigation</th><td>891</td></tr>\n",
       "\t<tr><th scope=row>disease_detail</th><td>  8</td></tr>\n",
       "</tbody>\n",
       "</table>\n",
       "</dd>\n",
       "\t<dt>$MDACC</dt>\n",
       "\t\t<dd><table class=\"dataframe\">\n",
       "<caption>A data.frame: 29 × 1</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>.</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>...1</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>contributor</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>file_name</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>patient_id</th><td> 12</td></tr>\n",
       "\t<tr><th scope=row>model_id</th><td>264</td></tr>\n",
       "\t<tr><th scope=row>tumor_id</th><td>418</td></tr>\n",
       "\t<tr><th scope=row>case_id</th><td> 12</td></tr>\n",
       "\t<tr><th scope=row>sample_id</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>sample_type</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>experimental_strategy</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>gender</th><td> 12</td></tr>\n",
       "\t<tr><th scope=row>paired_end</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>capture_kit</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>capture_assembly</th><td> 12</td></tr>\n",
       "\t<tr><th scope=row>availability</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>public</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>access_level</th><td>406</td></tr>\n",
       "\t<tr><th scope=row>created_datetime</th><td>418</td></tr>\n",
       "\t<tr><th scope=row>data_category</th><td>406</td></tr>\n",
       "\t<tr><th scope=row>data_format</th><td>406</td></tr>\n",
       "\t<tr><th scope=row>data_type</th><td>406</td></tr>\n",
       "\t<tr><th scope=row>platform</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>is_ffpe</th><td>418</td></tr>\n",
       "\t<tr><th scope=row>file_size</th><td>418</td></tr>\n",
       "\t<tr><th scope=row>passage</th><td>272</td></tr>\n",
       "\t<tr><th scope=row>disease_type</th><td> 12</td></tr>\n",
       "\t<tr><th scope=row>comment</th><td>404</td></tr>\n",
       "\t<tr><th scope=row>investigation</th><td>418</td></tr>\n",
       "\t<tr><th scope=row>disease_detail</th><td> 12</td></tr>\n",
       "</tbody>\n",
       "</table>\n",
       "</dd>\n",
       "\t<dt>$`UC Davis`</dt>\n",
       "\t\t<dd><table class=\"dataframe\">\n",
       "<caption>A data.frame: 29 × 1</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>.</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>...1</th><td> 0</td></tr>\n",
       "\t<tr><th scope=row>contributor</th><td> 0</td></tr>\n",
       "\t<tr><th scope=row>file_name</th><td> 0</td></tr>\n",
       "\t<tr><th scope=row>patient_id</th><td> 0</td></tr>\n",
       "\t<tr><th scope=row>model_id</th><td> 0</td></tr>\n",
       "\t<tr><th scope=row>tumor_id</th><td> 0</td></tr>\n",
       "\t<tr><th scope=row>case_id</th><td> 0</td></tr>\n",
       "\t<tr><th scope=row>sample_id</th><td> 0</td></tr>\n",
       "\t<tr><th scope=row>sample_type</th><td> 0</td></tr>\n",
       "\t<tr><th scope=row>experimental_strategy</th><td> 0</td></tr>\n",
       "\t<tr><th scope=row>gender</th><td> 0</td></tr>\n",
       "\t<tr><th scope=row>paired_end</th><td> 0</td></tr>\n",
       "\t<tr><th scope=row>capture_kit</th><td> 0</td></tr>\n",
       "\t<tr><th scope=row>capture_assembly</th><td>48</td></tr>\n",
       "\t<tr><th scope=row>availability</th><td> 0</td></tr>\n",
       "\t<tr><th scope=row>public</th><td>48</td></tr>\n",
       "\t<tr><th scope=row>access_level</th><td>48</td></tr>\n",
       "\t<tr><th scope=row>created_datetime</th><td>48</td></tr>\n",
       "\t<tr><th scope=row>data_category</th><td>48</td></tr>\n",
       "\t<tr><th scope=row>data_format</th><td> 0</td></tr>\n",
       "\t<tr><th scope=row>data_type</th><td> 0</td></tr>\n",
       "\t<tr><th scope=row>platform</th><td> 0</td></tr>\n",
       "\t<tr><th scope=row>is_ffpe</th><td>48</td></tr>\n",
       "\t<tr><th scope=row>file_size</th><td>48</td></tr>\n",
       "\t<tr><th scope=row>passage</th><td> 0</td></tr>\n",
       "\t<tr><th scope=row>disease_type</th><td> 0</td></tr>\n",
       "\t<tr><th scope=row>comment</th><td>48</td></tr>\n",
       "\t<tr><th scope=row>investigation</th><td>48</td></tr>\n",
       "\t<tr><th scope=row>disease_detail</th><td> 0</td></tr>\n",
       "</tbody>\n",
       "</table>\n",
       "</dd>\n",
       "\t<dt>$WISTAR</dt>\n",
       "\t\t<dd><table class=\"dataframe\">\n",
       "<caption>A data.frame: 29 × 1</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>.</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>...1</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>contributor</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>file_name</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>patient_id</th><td> 12</td></tr>\n",
       "\t<tr><th scope=row>model_id</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>tumor_id</th><td>382</td></tr>\n",
       "\t<tr><th scope=row>case_id</th><td>382</td></tr>\n",
       "\t<tr><th scope=row>sample_id</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>sample_type</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>experimental_strategy</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>gender</th><td>382</td></tr>\n",
       "\t<tr><th scope=row>paired_end</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>capture_kit</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>capture_assembly</th><td>382</td></tr>\n",
       "\t<tr><th scope=row>availability</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>public</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>access_level</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>created_datetime</th><td>370</td></tr>\n",
       "\t<tr><th scope=row>data_category</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>data_format</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>data_type</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>platform</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>is_ffpe</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>file_size</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>passage</th><td> 12</td></tr>\n",
       "\t<tr><th scope=row>disease_type</th><td>382</td></tr>\n",
       "\t<tr><th scope=row>comment</th><td>382</td></tr>\n",
       "\t<tr><th scope=row>investigation</th><td>382</td></tr>\n",
       "\t<tr><th scope=row>disease_detail</th><td>382</td></tr>\n",
       "</tbody>\n",
       "</table>\n",
       "</dd>\n",
       "\t<dt>$WUSTL</dt>\n",
       "\t\t<dd><table class=\"dataframe\">\n",
       "<caption>A data.frame: 29 × 1</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>.</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>...1</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>contributor</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>file_name</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>patient_id</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>model_id</th><td>360</td></tr>\n",
       "\t<tr><th scope=row>tumor_id</th><td>360</td></tr>\n",
       "\t<tr><th scope=row>case_id</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>sample_id</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>sample_type</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>experimental_strategy</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>gender</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>paired_end</th><td>  9</td></tr>\n",
       "\t<tr><th scope=row>capture_kit</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>capture_assembly</th><td>872</td></tr>\n",
       "\t<tr><th scope=row>availability</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>public</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>access_level</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>created_datetime</th><td>860</td></tr>\n",
       "\t<tr><th scope=row>data_category</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>data_format</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>data_type</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>platform</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>is_ffpe</th><td>875</td></tr>\n",
       "\t<tr><th scope=row>file_size</th><td>  0</td></tr>\n",
       "\t<tr><th scope=row>passage</th><td>360</td></tr>\n",
       "\t<tr><th scope=row>disease_type</th><td> 15</td></tr>\n",
       "\t<tr><th scope=row>comment</th><td>875</td></tr>\n",
       "\t<tr><th scope=row>investigation</th><td>875</td></tr>\n",
       "\t<tr><th scope=row>disease_detail</th><td> 15</td></tr>\n",
       "</tbody>\n",
       "</table>\n",
       "</dd>\n",
       "</dl>\n"
      ],
      "text/latex": [
       "\\begin{description}\n",
       "\\item[\\$BCM] A data.frame: 29 × 1\n",
       "\\begin{tabular}{r|l}\n",
       "  & .\\\\\n",
       "  & <dbl>\\\\\n",
       "\\hline\n",
       "\t...1 &   0\\\\\n",
       "\tcontributor &   0\\\\\n",
       "\tfile\\_name &   0\\\\\n",
       "\tpatient\\_id &   2\\\\\n",
       "\tmodel\\_id &   0\\\\\n",
       "\ttumor\\_id &  16\\\\\n",
       "\tcase\\_id & 208\\\\\n",
       "\tsample\\_id &   0\\\\\n",
       "\tsample\\_type &   0\\\\\n",
       "\texperimental\\_strategy &   0\\\\\n",
       "\tgender &  42\\\\\n",
       "\tpaired\\_end &   0\\\\\n",
       "\tcapture\\_kit & 124\\\\\n",
       "\tcapture\\_assembly & 208\\\\\n",
       "\tavailability &   0\\\\\n",
       "\tpublic &   0\\\\\n",
       "\taccess\\_level &   0\\\\\n",
       "\tcreated\\_datetime & 124\\\\\n",
       "\tdata\\_category &   0\\\\\n",
       "\tdata\\_format &   0\\\\\n",
       "\tdata\\_type &   0\\\\\n",
       "\tplatform & 124\\\\\n",
       "\tis\\_ffpe & 124\\\\\n",
       "\tfile\\_size &  46\\\\\n",
       "\tpassage & 208\\\\\n",
       "\tdisease\\_type &  98\\\\\n",
       "\tcomment & 208\\\\\n",
       "\tinvestigation & 208\\\\\n",
       "\tdisease\\_detail &  98\\\\\n",
       "\\end{tabular}\n",
       "\n",
       "\\item[\\$HCI] A data.frame: 29 × 1\n",
       "\\begin{tabular}{r|l}\n",
       "  & .\\\\\n",
       "  & <dbl>\\\\\n",
       "\\hline\n",
       "\t...1 &   0\\\\\n",
       "\tcontributor &   0\\\\\n",
       "\tfile\\_name &   0\\\\\n",
       "\tpatient\\_id &   0\\\\\n",
       "\tmodel\\_id & 891\\\\\n",
       "\ttumor\\_id & 891\\\\\n",
       "\tcase\\_id &   0\\\\\n",
       "\tsample\\_id &   0\\\\\n",
       "\tsample\\_type &   0\\\\\n",
       "\texperimental\\_strategy &   0\\\\\n",
       "\tgender &   0\\\\\n",
       "\tpaired\\_end &   8\\\\\n",
       "\tcapture\\_kit & 697\\\\\n",
       "\tcapture\\_assembly & 683\\\\\n",
       "\tavailability &   0\\\\\n",
       "\tpublic &   0\\\\\n",
       "\taccess\\_level & 891\\\\\n",
       "\tcreated\\_datetime & 891\\\\\n",
       "\tdata\\_category & 891\\\\\n",
       "\tdata\\_format & 891\\\\\n",
       "\tdata\\_type & 891\\\\\n",
       "\tplatform &   0\\\\\n",
       "\tis\\_ffpe & 891\\\\\n",
       "\tfile\\_size & 891\\\\\n",
       "\tpassage & 779\\\\\n",
       "\tdisease\\_type &   8\\\\\n",
       "\tcomment & 859\\\\\n",
       "\tinvestigation & 891\\\\\n",
       "\tdisease\\_detail &   8\\\\\n",
       "\\end{tabular}\n",
       "\n",
       "\\item[\\$MDACC] A data.frame: 29 × 1\n",
       "\\begin{tabular}{r|l}\n",
       "  & .\\\\\n",
       "  & <dbl>\\\\\n",
       "\\hline\n",
       "\t...1 &   0\\\\\n",
       "\tcontributor &   0\\\\\n",
       "\tfile\\_name &   0\\\\\n",
       "\tpatient\\_id &  12\\\\\n",
       "\tmodel\\_id & 264\\\\\n",
       "\ttumor\\_id & 418\\\\\n",
       "\tcase\\_id &  12\\\\\n",
       "\tsample\\_id &   0\\\\\n",
       "\tsample\\_type &   0\\\\\n",
       "\texperimental\\_strategy &   0\\\\\n",
       "\tgender &  12\\\\\n",
       "\tpaired\\_end &   0\\\\\n",
       "\tcapture\\_kit &   0\\\\\n",
       "\tcapture\\_assembly &  12\\\\\n",
       "\tavailability &   0\\\\\n",
       "\tpublic &   0\\\\\n",
       "\taccess\\_level & 406\\\\\n",
       "\tcreated\\_datetime & 418\\\\\n",
       "\tdata\\_category & 406\\\\\n",
       "\tdata\\_format & 406\\\\\n",
       "\tdata\\_type & 406\\\\\n",
       "\tplatform &   0\\\\\n",
       "\tis\\_ffpe & 418\\\\\n",
       "\tfile\\_size & 418\\\\\n",
       "\tpassage & 272\\\\\n",
       "\tdisease\\_type &  12\\\\\n",
       "\tcomment & 404\\\\\n",
       "\tinvestigation & 418\\\\\n",
       "\tdisease\\_detail &  12\\\\\n",
       "\\end{tabular}\n",
       "\n",
       "\\item[\\$`UC Davis`] A data.frame: 29 × 1\n",
       "\\begin{tabular}{r|l}\n",
       "  & .\\\\\n",
       "  & <dbl>\\\\\n",
       "\\hline\n",
       "\t...1 &  0\\\\\n",
       "\tcontributor &  0\\\\\n",
       "\tfile\\_name &  0\\\\\n",
       "\tpatient\\_id &  0\\\\\n",
       "\tmodel\\_id &  0\\\\\n",
       "\ttumor\\_id &  0\\\\\n",
       "\tcase\\_id &  0\\\\\n",
       "\tsample\\_id &  0\\\\\n",
       "\tsample\\_type &  0\\\\\n",
       "\texperimental\\_strategy &  0\\\\\n",
       "\tgender &  0\\\\\n",
       "\tpaired\\_end &  0\\\\\n",
       "\tcapture\\_kit &  0\\\\\n",
       "\tcapture\\_assembly & 48\\\\\n",
       "\tavailability &  0\\\\\n",
       "\tpublic & 48\\\\\n",
       "\taccess\\_level & 48\\\\\n",
       "\tcreated\\_datetime & 48\\\\\n",
       "\tdata\\_category & 48\\\\\n",
       "\tdata\\_format &  0\\\\\n",
       "\tdata\\_type &  0\\\\\n",
       "\tplatform &  0\\\\\n",
       "\tis\\_ffpe & 48\\\\\n",
       "\tfile\\_size & 48\\\\\n",
       "\tpassage &  0\\\\\n",
       "\tdisease\\_type &  0\\\\\n",
       "\tcomment & 48\\\\\n",
       "\tinvestigation & 48\\\\\n",
       "\tdisease\\_detail &  0\\\\\n",
       "\\end{tabular}\n",
       "\n",
       "\\item[\\$WISTAR] A data.frame: 29 × 1\n",
       "\\begin{tabular}{r|l}\n",
       "  & .\\\\\n",
       "  & <dbl>\\\\\n",
       "\\hline\n",
       "\t...1 &   0\\\\\n",
       "\tcontributor &   0\\\\\n",
       "\tfile\\_name &   0\\\\\n",
       "\tpatient\\_id &  12\\\\\n",
       "\tmodel\\_id &   0\\\\\n",
       "\ttumor\\_id & 382\\\\\n",
       "\tcase\\_id & 382\\\\\n",
       "\tsample\\_id &   0\\\\\n",
       "\tsample\\_type &   0\\\\\n",
       "\texperimental\\_strategy &   0\\\\\n",
       "\tgender & 382\\\\\n",
       "\tpaired\\_end &   0\\\\\n",
       "\tcapture\\_kit &   0\\\\\n",
       "\tcapture\\_assembly & 382\\\\\n",
       "\tavailability &   0\\\\\n",
       "\tpublic &   0\\\\\n",
       "\taccess\\_level &   0\\\\\n",
       "\tcreated\\_datetime & 370\\\\\n",
       "\tdata\\_category &   0\\\\\n",
       "\tdata\\_format &   0\\\\\n",
       "\tdata\\_type &   0\\\\\n",
       "\tplatform &   0\\\\\n",
       "\tis\\_ffpe &   0\\\\\n",
       "\tfile\\_size &   0\\\\\n",
       "\tpassage &  12\\\\\n",
       "\tdisease\\_type & 382\\\\\n",
       "\tcomment & 382\\\\\n",
       "\tinvestigation & 382\\\\\n",
       "\tdisease\\_detail & 382\\\\\n",
       "\\end{tabular}\n",
       "\n",
       "\\item[\\$WUSTL] A data.frame: 29 × 1\n",
       "\\begin{tabular}{r|l}\n",
       "  & .\\\\\n",
       "  & <dbl>\\\\\n",
       "\\hline\n",
       "\t...1 &   0\\\\\n",
       "\tcontributor &   0\\\\\n",
       "\tfile\\_name &   0\\\\\n",
       "\tpatient\\_id &   0\\\\\n",
       "\tmodel\\_id & 360\\\\\n",
       "\ttumor\\_id & 360\\\\\n",
       "\tcase\\_id &   0\\\\\n",
       "\tsample\\_id &   0\\\\\n",
       "\tsample\\_type &   0\\\\\n",
       "\texperimental\\_strategy &   0\\\\\n",
       "\tgender &   0\\\\\n",
       "\tpaired\\_end &   9\\\\\n",
       "\tcapture\\_kit &   0\\\\\n",
       "\tcapture\\_assembly & 872\\\\\n",
       "\tavailability &   0\\\\\n",
       "\tpublic &   0\\\\\n",
       "\taccess\\_level &   0\\\\\n",
       "\tcreated\\_datetime & 860\\\\\n",
       "\tdata\\_category &   0\\\\\n",
       "\tdata\\_format &   0\\\\\n",
       "\tdata\\_type &   0\\\\\n",
       "\tplatform &   0\\\\\n",
       "\tis\\_ffpe & 875\\\\\n",
       "\tfile\\_size &   0\\\\\n",
       "\tpassage & 360\\\\\n",
       "\tdisease\\_type &  15\\\\\n",
       "\tcomment & 875\\\\\n",
       "\tinvestigation & 875\\\\\n",
       "\tdisease\\_detail &  15\\\\\n",
       "\\end{tabular}\n",
       "\n",
       "\\end{description}\n"
      ],
      "text/markdown": [
       "$BCM\n",
       ":   \n",
       "A data.frame: 29 × 1\n",
       "\n",
       "| <!--/--> | . &lt;dbl&gt; |\n",
       "|---|---|\n",
       "| ...1 |   0 |\n",
       "| contributor |   0 |\n",
       "| file_name |   0 |\n",
       "| patient_id |   2 |\n",
       "| model_id |   0 |\n",
       "| tumor_id |  16 |\n",
       "| case_id | 208 |\n",
       "| sample_id |   0 |\n",
       "| sample_type |   0 |\n",
       "| experimental_strategy |   0 |\n",
       "| gender |  42 |\n",
       "| paired_end |   0 |\n",
       "| capture_kit | 124 |\n",
       "| capture_assembly | 208 |\n",
       "| availability |   0 |\n",
       "| public |   0 |\n",
       "| access_level |   0 |\n",
       "| created_datetime | 124 |\n",
       "| data_category |   0 |\n",
       "| data_format |   0 |\n",
       "| data_type |   0 |\n",
       "| platform | 124 |\n",
       "| is_ffpe | 124 |\n",
       "| file_size |  46 |\n",
       "| passage | 208 |\n",
       "| disease_type |  98 |\n",
       "| comment | 208 |\n",
       "| investigation | 208 |\n",
       "| disease_detail |  98 |\n",
       "\n",
       "\n",
       "$HCI\n",
       ":   \n",
       "A data.frame: 29 × 1\n",
       "\n",
       "| <!--/--> | . &lt;dbl&gt; |\n",
       "|---|---|\n",
       "| ...1 |   0 |\n",
       "| contributor |   0 |\n",
       "| file_name |   0 |\n",
       "| patient_id |   0 |\n",
       "| model_id | 891 |\n",
       "| tumor_id | 891 |\n",
       "| case_id |   0 |\n",
       "| sample_id |   0 |\n",
       "| sample_type |   0 |\n",
       "| experimental_strategy |   0 |\n",
       "| gender |   0 |\n",
       "| paired_end |   8 |\n",
       "| capture_kit | 697 |\n",
       "| capture_assembly | 683 |\n",
       "| availability |   0 |\n",
       "| public |   0 |\n",
       "| access_level | 891 |\n",
       "| created_datetime | 891 |\n",
       "| data_category | 891 |\n",
       "| data_format | 891 |\n",
       "| data_type | 891 |\n",
       "| platform |   0 |\n",
       "| is_ffpe | 891 |\n",
       "| file_size | 891 |\n",
       "| passage | 779 |\n",
       "| disease_type |   8 |\n",
       "| comment | 859 |\n",
       "| investigation | 891 |\n",
       "| disease_detail |   8 |\n",
       "\n",
       "\n",
       "$MDACC\n",
       ":   \n",
       "A data.frame: 29 × 1\n",
       "\n",
       "| <!--/--> | . &lt;dbl&gt; |\n",
       "|---|---|\n",
       "| ...1 |   0 |\n",
       "| contributor |   0 |\n",
       "| file_name |   0 |\n",
       "| patient_id |  12 |\n",
       "| model_id | 264 |\n",
       "| tumor_id | 418 |\n",
       "| case_id |  12 |\n",
       "| sample_id |   0 |\n",
       "| sample_type |   0 |\n",
       "| experimental_strategy |   0 |\n",
       "| gender |  12 |\n",
       "| paired_end |   0 |\n",
       "| capture_kit |   0 |\n",
       "| capture_assembly |  12 |\n",
       "| availability |   0 |\n",
       "| public |   0 |\n",
       "| access_level | 406 |\n",
       "| created_datetime | 418 |\n",
       "| data_category | 406 |\n",
       "| data_format | 406 |\n",
       "| data_type | 406 |\n",
       "| platform |   0 |\n",
       "| is_ffpe | 418 |\n",
       "| file_size | 418 |\n",
       "| passage | 272 |\n",
       "| disease_type |  12 |\n",
       "| comment | 404 |\n",
       "| investigation | 418 |\n",
       "| disease_detail |  12 |\n",
       "\n",
       "\n",
       "$`UC Davis`\n",
       ":   \n",
       "A data.frame: 29 × 1\n",
       "\n",
       "| <!--/--> | . &lt;dbl&gt; |\n",
       "|---|---|\n",
       "| ...1 |  0 |\n",
       "| contributor |  0 |\n",
       "| file_name |  0 |\n",
       "| patient_id |  0 |\n",
       "| model_id |  0 |\n",
       "| tumor_id |  0 |\n",
       "| case_id |  0 |\n",
       "| sample_id |  0 |\n",
       "| sample_type |  0 |\n",
       "| experimental_strategy |  0 |\n",
       "| gender |  0 |\n",
       "| paired_end |  0 |\n",
       "| capture_kit |  0 |\n",
       "| capture_assembly | 48 |\n",
       "| availability |  0 |\n",
       "| public | 48 |\n",
       "| access_level | 48 |\n",
       "| created_datetime | 48 |\n",
       "| data_category | 48 |\n",
       "| data_format |  0 |\n",
       "| data_type |  0 |\n",
       "| platform |  0 |\n",
       "| is_ffpe | 48 |\n",
       "| file_size | 48 |\n",
       "| passage |  0 |\n",
       "| disease_type |  0 |\n",
       "| comment | 48 |\n",
       "| investigation | 48 |\n",
       "| disease_detail |  0 |\n",
       "\n",
       "\n",
       "$WISTAR\n",
       ":   \n",
       "A data.frame: 29 × 1\n",
       "\n",
       "| <!--/--> | . &lt;dbl&gt; |\n",
       "|---|---|\n",
       "| ...1 |   0 |\n",
       "| contributor |   0 |\n",
       "| file_name |   0 |\n",
       "| patient_id |  12 |\n",
       "| model_id |   0 |\n",
       "| tumor_id | 382 |\n",
       "| case_id | 382 |\n",
       "| sample_id |   0 |\n",
       "| sample_type |   0 |\n",
       "| experimental_strategy |   0 |\n",
       "| gender | 382 |\n",
       "| paired_end |   0 |\n",
       "| capture_kit |   0 |\n",
       "| capture_assembly | 382 |\n",
       "| availability |   0 |\n",
       "| public |   0 |\n",
       "| access_level |   0 |\n",
       "| created_datetime | 370 |\n",
       "| data_category |   0 |\n",
       "| data_format |   0 |\n",
       "| data_type |   0 |\n",
       "| platform |   0 |\n",
       "| is_ffpe |   0 |\n",
       "| file_size |   0 |\n",
       "| passage |  12 |\n",
       "| disease_type | 382 |\n",
       "| comment | 382 |\n",
       "| investigation | 382 |\n",
       "| disease_detail | 382 |\n",
       "\n",
       "\n",
       "$WUSTL\n",
       ":   \n",
       "A data.frame: 29 × 1\n",
       "\n",
       "| <!--/--> | . &lt;dbl&gt; |\n",
       "|---|---|\n",
       "| ...1 |   0 |\n",
       "| contributor |   0 |\n",
       "| file_name |   0 |\n",
       "| patient_id |   0 |\n",
       "| model_id | 360 |\n",
       "| tumor_id | 360 |\n",
       "| case_id |   0 |\n",
       "| sample_id |   0 |\n",
       "| sample_type |   0 |\n",
       "| experimental_strategy |   0 |\n",
       "| gender |   0 |\n",
       "| paired_end |   9 |\n",
       "| capture_kit |   0 |\n",
       "| capture_assembly | 872 |\n",
       "| availability |   0 |\n",
       "| public |   0 |\n",
       "| access_level |   0 |\n",
       "| created_datetime | 860 |\n",
       "| data_category |   0 |\n",
       "| data_format |   0 |\n",
       "| data_type |   0 |\n",
       "| platform |   0 |\n",
       "| is_ffpe | 875 |\n",
       "| file_size |   0 |\n",
       "| passage | 360 |\n",
       "| disease_type |  15 |\n",
       "| comment | 875 |\n",
       "| investigation | 875 |\n",
       "| disease_detail |  15 |\n",
       "\n",
       "\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "$BCM\n",
       "                        .\n",
       "...1                    0\n",
       "contributor             0\n",
       "file_name               0\n",
       "patient_id              2\n",
       "model_id                0\n",
       "tumor_id               16\n",
       "case_id               208\n",
       "sample_id               0\n",
       "sample_type             0\n",
       "experimental_strategy   0\n",
       "gender                 42\n",
       "paired_end              0\n",
       "capture_kit           124\n",
       "capture_assembly      208\n",
       "availability            0\n",
       "public                  0\n",
       "access_level            0\n",
       "created_datetime      124\n",
       "data_category           0\n",
       "data_format             0\n",
       "data_type               0\n",
       "platform              124\n",
       "is_ffpe               124\n",
       "file_size              46\n",
       "passage               208\n",
       "disease_type           98\n",
       "comment               208\n",
       "investigation         208\n",
       "disease_detail         98\n",
       "\n",
       "$HCI\n",
       "                        .\n",
       "...1                    0\n",
       "contributor             0\n",
       "file_name               0\n",
       "patient_id              0\n",
       "model_id              891\n",
       "tumor_id              891\n",
       "case_id                 0\n",
       "sample_id               0\n",
       "sample_type             0\n",
       "experimental_strategy   0\n",
       "gender                  0\n",
       "paired_end              8\n",
       "capture_kit           697\n",
       "capture_assembly      683\n",
       "availability            0\n",
       "public                  0\n",
       "access_level          891\n",
       "created_datetime      891\n",
       "data_category         891\n",
       "data_format           891\n",
       "data_type             891\n",
       "platform                0\n",
       "is_ffpe               891\n",
       "file_size             891\n",
       "passage               779\n",
       "disease_type            8\n",
       "comment               859\n",
       "investigation         891\n",
       "disease_detail          8\n",
       "\n",
       "$MDACC\n",
       "                        .\n",
       "...1                    0\n",
       "contributor             0\n",
       "file_name               0\n",
       "patient_id             12\n",
       "model_id              264\n",
       "tumor_id              418\n",
       "case_id                12\n",
       "sample_id               0\n",
       "sample_type             0\n",
       "experimental_strategy   0\n",
       "gender                 12\n",
       "paired_end              0\n",
       "capture_kit             0\n",
       "capture_assembly       12\n",
       "availability            0\n",
       "public                  0\n",
       "access_level          406\n",
       "created_datetime      418\n",
       "data_category         406\n",
       "data_format           406\n",
       "data_type             406\n",
       "platform                0\n",
       "is_ffpe               418\n",
       "file_size             418\n",
       "passage               272\n",
       "disease_type           12\n",
       "comment               404\n",
       "investigation         418\n",
       "disease_detail         12\n",
       "\n",
       "$`UC Davis`\n",
       "                       .\n",
       "...1                   0\n",
       "contributor            0\n",
       "file_name              0\n",
       "patient_id             0\n",
       "model_id               0\n",
       "tumor_id               0\n",
       "case_id                0\n",
       "sample_id              0\n",
       "sample_type            0\n",
       "experimental_strategy  0\n",
       "gender                 0\n",
       "paired_end             0\n",
       "capture_kit            0\n",
       "capture_assembly      48\n",
       "availability           0\n",
       "public                48\n",
       "access_level          48\n",
       "created_datetime      48\n",
       "data_category         48\n",
       "data_format            0\n",
       "data_type              0\n",
       "platform               0\n",
       "is_ffpe               48\n",
       "file_size             48\n",
       "passage                0\n",
       "disease_type           0\n",
       "comment               48\n",
       "investigation         48\n",
       "disease_detail         0\n",
       "\n",
       "$WISTAR\n",
       "                        .\n",
       "...1                    0\n",
       "contributor             0\n",
       "file_name               0\n",
       "patient_id             12\n",
       "model_id                0\n",
       "tumor_id              382\n",
       "case_id               382\n",
       "sample_id               0\n",
       "sample_type             0\n",
       "experimental_strategy   0\n",
       "gender                382\n",
       "paired_end              0\n",
       "capture_kit             0\n",
       "capture_assembly      382\n",
       "availability            0\n",
       "public                  0\n",
       "access_level            0\n",
       "created_datetime      370\n",
       "data_category           0\n",
       "data_format             0\n",
       "data_type               0\n",
       "platform                0\n",
       "is_ffpe                 0\n",
       "file_size               0\n",
       "passage                12\n",
       "disease_type          382\n",
       "comment               382\n",
       "investigation         382\n",
       "disease_detail        382\n",
       "\n",
       "$WUSTL\n",
       "                        .\n",
       "...1                    0\n",
       "contributor             0\n",
       "file_name               0\n",
       "patient_id              0\n",
       "model_id              360\n",
       "tumor_id              360\n",
       "case_id                 0\n",
       "sample_id               0\n",
       "sample_type             0\n",
       "experimental_strategy   0\n",
       "gender                  0\n",
       "paired_end              9\n",
       "capture_kit             0\n",
       "capture_assembly      872\n",
       "availability            0\n",
       "public                  0\n",
       "access_level            0\n",
       "created_datetime      860\n",
       "data_category           0\n",
       "data_format             0\n",
       "data_type               0\n",
       "platform                0\n",
       "is_ffpe               875\n",
       "file_size               0\n",
       "passage               360\n",
       "disease_type           15\n",
       "comment               875\n",
       "investigation         875\n",
       "disease_detail         15\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Get the number of missing metadata fields from the original metadata\n",
    "\n",
    "original_pdtc_metadata %>% map(summarize_missing_fields)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4abc8dba-eff3-4415-bd20-0755a69d8c39",
   "metadata": {},
   "source": [
    "At this point we only have 80 missing disease information (vs over 500 originally) and 26 missing capture_kit info. Passage is problematic. Some that are missing gender/sex information are probably ones where the patient did not provide it. Also, all unmerged lanes and non-fastqs have been removed and replaced with their merged or fastq-converted counterparts. Unnecessary metadata fields like `created_datetime` and `file_size` were removed. `disease_type` was renamed to `body_location` and `disease_detail` changed to `disease_type` to match the fields for the PDMR.\n",
    "\n",
    "As before, library prep kits for RNAseq are also included in the `capture_kit` column.\n",
    "\n",
    "Other changes include using \"Unknown\" for missing data in most fields, and NA where they do not apply. Such as primary tumors getting NA in passage or model ID, but samples without known `disease_type` getting \"Unknown.\" All files were set to \"Controlled\" access unless already public by the PDTC according to their metadata.\n",
    "\n",
    "Many other missing fields here and there pulled from model info, metadata included in the center's dbGaP submission, or in some cases from the metadata on the CGC.\n",
    "\n",
    "This metadata can now also be applied to all these files on the CGC. That will also indicate files on the CGC that are not included in this list, if any.\n",
    "\n",
    "At this time, I don't think it will be possible to get the remaining missing info without going back to the individual PDTCs for that information."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "eefb9006-5a50-4b5f-9012-2cd81da84dd2",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
